---
title: "Use gghic to Visualise Genomic Interaction Data"
author: "Minghao Jiang"
date: "`r Sys.Date()`"
output:
  rmarkdown::html_document:
    toc: true
    toc_float: true
    number_sections: false
    center: false
    theme: default
    highlight: pygments
    css: style.css
vignette: >
  %\VignetteIndexEntry{gghic}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
  %\VignetteKeywords{genomics, visualisation, Hi-C}
  %\VignetteAbstract{This vignette demonstrates how to use the gghic package to visualise genomic interaction data.}
editor_options:
  markdown:
    wrap: 80
    code_folding: true
    scroll_past_end: true
    tab_size: 2
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  tidy = FALSE,
  cache = FALSE,
  dev = "png"
)
```

## Introduction

gghic is an R package designed for flexible and publication-ready visualization of 3D genome organization data, including Hi-C, topologically associating domains (TADs), chromatin loops, gene annotation, and more. This tutorial demonstrates how to use gghic to visualize various types of genomic interaction data.

## Setup

### Load Required Packages

```{r setup}
load_pkg <- function(pkgs) {
  for (pkg in pkgs) suppressMessages(require(pkg, character.only = TRUE))
}

load_pkg(
  c(
    "gghic", "ggplot2", "tibble", "scales", "dplyr", "rappdirs",
    "tidyr", "HiCExperiment", "InteractionSet", "glue"
  )
)
```

### Download Example data

Example data files are hosted at gghic-data to keep the package lightweight.

```{r message=FALSE, warning=FALSE}
download_example_files <- function(check_exists = FALSE) {
  dir_cache <- rappdirs::user_cache_dir(appname = "gghic")

  files <- list(
    "chr4_11-100kb.cool" = "cooler/chr4_11-100kb.cool",
    "chr4_11-5kb.cool" = "cooler/chr4_11-5kb.cool",
    "track1.bigWig" = "bigwig/track1.bigWig",
    "track2.bigWig" = "bigwig/track2.bigWig",
    "gencode-chr4_11.gtf.gz" = "gtf/gencode-chr4_11.gtf.gz",
    "TADs_500kb-chr4_11.tsv" = "tad/TADs_500kb-chr4_11.tsv",
    "loops-chr4_11.txt" = "loop/loops-chr4_11.txt",
    "gis_hic.rds" = "multiway/gis_hic.rds",
    "concatemers.rds" = "multiway/concatemers.rds"
  )

  url_base <- paste0(
    "https://raw.githubusercontent.com/",
    "mhjiang97/gghic-data/refs/heads/master/"
  )

  for (file_name in names(files)) {
    file_path <- file.path(dir_cache, file_name)
    if (check_exists && file.exists(file_path)) next
    utils::download.file(paste0(url_base, files[[file_name]]), file_path)
  }
}
```

```{r message=FALSE, warning=FALSE}
download_example_files()

dir_cache_gghic <- rappdirs::user_cache_dir(appname = "gghic")

path_cf_100 <- file.path(dir_cache_gghic, "chr4_11-100kb.cool")
path_cf_5 <- file.path(dir_cache_gghic, "chr4_11-5kb.cool")
path_gtf <- file.path(dir_cache_gghic, "gencode-chr4_11.gtf.gz")
paths_track <- file.path(dir_cache_gghic, glue::glue("track{1:2}.bigWig"))
path_tad <- file.path(dir_cache_gghic, "TADs_500kb-chr4_11.tsv")
path_loop <- file.path(dir_cache_gghic, "loops-chr4_11.txt")
path_gis_hic <- file.path(dir_cache_gghic, "gis_hic.rds")
path_concatemers <- file.path(dir_cache_gghic, "concatemers.rds")
```

## Data Preparation

### Import Hi-C Data

```{r}
hic_100 <- path_cf_100 |>
  HiCExperiment::CoolFile() |>
  HiCExperiment::import()

hic_5 <- path_cf_5 |>
  HiCExperiment::CoolFile() |>
  HiCExperiment::import()
```

### Prepare Data for Visualisation

```{r}
scale_data <- function(data, score_column = "balanced", scale_method = log10) {
  if (inherits(data, "HiCExperiment")) {
    gis <- InteractionSet::interactions(data)
  } else if (inherits(data, "GInteractions")) {
    gis <- data
  } else {
    stop("Input data must be a HiCExperiment or GInteractions object.")
  }

  x <- gis |>
    tibble::as_tibble() |>
    dplyr::mutate(score = scale_method(.data[[score_column]])) |>
    dplyr::filter(
      InteractionSet::pairdist(gis) != 0,
      !is.na(InteractionSet::pairdist(gis) != 0)
    ) |>
    dplyr::filter(!is.na(score), !is.infinite(score)) |>
    dplyr::mutate(score = scales::oob_squish(score, c(min(score), max(score))))

  x
}
```

```{r}
x_100 <- scale_data(hic_100)
x_5 <- scale_data(hic_5)
```

## Visualisations

### Hi-C Heatmap

`geom_hic()` is the main function to visualise Hi-C/-like data. It requires the
following aesthetics: `seqnames1`, `start1`, `end1`, `seqnames2`, `start2`,
`end2`, and `fill`.

```{r}
p <- x_100 |>
  dplyr::filter(seqnames1 == "chr11", seqnames2 == "chr11") |>
  tidyr::drop_na(score) |>
  ggplot2::ggplot(
    ggplot2::aes(
      seqnames1 = seqnames1, start1 = start1, end1 = end1,
      seqnames2 = seqnames2, start2 = start2, end2 = end2,
      fill = score
    )
  ) +
  geom_hic()

p
```

`theme_hic()` is a theme function for `geom_hic()` that enhances the appearance
of the plot.

```{r}
p + theme_hic()
```

`gghic()` is a wrapper function that can take either a `HiCExperiment` object, a `GenomicInteractions` object, or a `tibble`/`data.frame` as input. (exactly the same as above).

```{r eval=FALSE, echo=TRUE}
hic_100["chr11"] |>
  gghic()
```

### Ideogram

`geom_ideogram()` can be used to add chromosome ideograms to the plot.

```{r}
x_100 |>
  dplyr::filter(seqnames1 == "chr11", seqnames2 == "chr11") |>
  ggplot2::ggplot(
    ggplot2::aes(
      seqnames1 = seqnames1, start1 = start1, end1 = end1,
      seqnames2 = seqnames2, start2 = start2, end2 = end2,
      fill = score
    )
  ) +
  geom_hic() +
  theme_hic() +
  geom_ideogram(
    genome = "hg19", highlight = FALSE, length_ratio = 0.7, fontsize = 8
  )
```

The wrapper function `gghic()` can also add ideograms to the plot. (exactly the same as above).

```{r eval=FALSE, echo=TRUE}
hic_100["chr11"] |>
  gghic(
    ideogram = TRUE, genome = "hg19", highlight = FALSE, length_ratio = 0.7,
    ideogram_fontsize = 8
  )
```

The plotted region can be highlighted on the ideogram by setting `highlight = TRUE`.

```{r}
hic_100["chr4:10000000-15000000"] |>
  gghic(
    ideogram = TRUE, genome = "hg19", highlight = TRUE, length_ratio = 0.7,
    ideogram_fontsize = 8
  )
```

### Gene Annotation

Track-based gene annotation can be added to the plot by using `geom_annotation()`.

```{r}
p <- x_5 |>
  dplyr::filter(
    seqnames1 == "chr11", seqnames2 == "chr11",
    center1 > 67000000, center1 < 67100000,
    center2 > 67000000, center2 < 67100000
  ) |>
  ggplot2::ggplot(
    ggplot2::aes(
      seqnames1 = seqnames1, start1 = start1, end1 = end1,
      seqnames2 = seqnames2, start2 = start2, end2 = end2,
      fill = score
    )
  ) +
  geom_hic() +
  theme_hic() +
  geom_ideogram(
    genome = "hg19", highlight = FALSE, length_ratio = 0.7, fontsize = 8
  )

p + geom_annotation(gtf_path = path_gtf, style = "basic", maxgap = 100000)
```

Some parameters can be set for `geom_annotation()`:

- `style` can be set to `"arrow"` to change the annotation style.
- `maxgap` is used to calculate how many non-overlapping tracks are needed for the annotation.
- `gene_symbols` can be used to specify genes to plot.

```{r}
p + geom_annotation(
  gtf_path = path_gtf, style = "arrow", maxgap = -1,
  gene_symbols = c("GRK2", "SSH3", "KDM2A")
)
```

`gghic()` can also add gene annotation to the plot (exactly the same as above).

```{r eval=FALSE, echo=TRUE}
hic_5["chr11:67000000-67100000"] |>
  gghic(
    ideogram = TRUE, genome = "hg19", highlight = FALSE, length_ratio = 0.7,
    ideogram_fontsize = 8,

    annotation = TRUE, gtf_path = path_gtf, style = "arrow", maxgap = -1,
    gene_symbols = c("GRK2", "SSH3", "KDM2A")
  )
```

### Chromatin Loops

Chromatin loops can be visualised using `geom_loop()` or `geom_loop2()`. The former requires a path to a loop file, while the latter requires a `tibble`/`data.frame` containing the loop data which enables easy customization by adding additional aesthetics.

```{r}
df_loop <- path_loop |>
  read.table(
    sep = "\t",
    col.names = c("seqnames1", "start1", "end1", "seqnames2", "start2", "end2")
  ) |>
  dplyr::filter(
    seqnames1 == "chr11", seqnames2 == "chr11",
    start1 > 61925000, end1 < 67480000,
    start2 > 61925000, end2 < 67480000
  )

keep <- sample(nrow(df_loop), 6)
df_loop <- df_loop |>
  dplyr::slice(keep) |>
  dplyr::mutate(
    sample = c(rep("A", 3), rep("B", 3))
  )

hic_5["chr11:61915000-65000000"] |>
  gghic() +
  geom_loop2(
    data = df_loop,
    aes(
      seqnames1 = seqnames1, start1 = start1, end1 = end1,
      seqnames2 = seqnames2, start2 = start2, end2 = end2,
      colour = sample
    ),
    stroke = 1
  )
```

### TADs

Triangulers indicating TADs can be added using `geom_tad()`.

```{r}
x_100 |>
  dplyr::filter(
    seqnames1 == "chr4", seqnames2 == "chr4",
    start1 > 60000000, end1 < 70000000,
    start2 > 60000000, end2 < 70000000
  ) |>
  gghic(
    tad = TRUE, tad_is_0based = TRUE, tad_path = path_tad, tad_colour = "#00ff83"
  )
```

### Multi-way contacts

Multi-way contacts from concatemers obtained from Pore-C data can be visualised using `geom_concatemer()` or `geom_concatemer2()`. The former requires a path to a concatemer file, while the latter requires a `tibble`/`data.frame` containing the concatemer data.

```{r}
gis <- path_gis_hic |>
  readRDS() |>
  scale_data("balanced", log10)

names_read <- c(
  "0c32bbce-24f8-4006-a75c-0e3504648971",
  "10ae8263-cebc-4cbc-94c1-2c1dc5f22101",
  "ced078ca-2d68-5217-a20e-7dc37dd2d786",
  "06063b40-0271-4f4e-993d-2f202028b697",
  "08bb8fe4-c54b-4a11-b9a0-c9eed2d18933"
)
df_concatemers <- path_concatemers |>
  readRDS() |>
  sort() |>
  tibble::as_tibble() |>
  dplyr::filter(read_name %in% names_read)

gghic(gis) +
  geom_concatemer2(
    data = df_concatemers,
    aes(
      seqnames = seqnames, start = start, end = end, read_group = read_name,
      colour = read_name
    ),
    check_concatemers = TRUE
  ) +
  ggplot2::theme(
    legend.key.size = ggplot2::unit(2, "mm"),
    legend.text = ggplot2::element_text(size = 4)
  )
```

### Visualisation of Interaction Data of Multiple Chromosomes

Set `draw_boundary` to `TRUE` to draw the boundary of the chromosomes. Set `expand_xaxis` to `TRUE` to expand the x-axis if the plot is crowded.

```{r}
x_100 |>
  gghic(
    draw_boundary = TRUE,
    expand_xaxis = TRUE, expand_left = 2000000, expand_right = 2000000
  )
```

Chromosome ideograms and gene tracks can also be added to the plot when visualising multiple chromosomes.

```{r}
p <- x_5 |>
  dplyr::filter(
    center1 > 10000000 & center1 < 11000000 &
      center2 > 10000000 & center2 < 11000000
  ) |>
  gghic(
    draw_boundary = TRUE,

    ideogram = TRUE, genome = "hg19", highlight = TRUE, ideogram_fontsize = 7,
    ideogram_width_ratio = 0.08,

    annotation = TRUE, include_ncrna = FALSE, gtf_path = path_gtf,
    style = "basic", maxgap = 100000, annotation_fontsize = 5,
    annotation_width_ratio = 0.05,

    expand_xaxis = TRUE
  )

p
```

### Other Data Tracks - `geom_track`

`geom_track()` can be used to add other data tracks to the plot.

```{r}
p + geom_track(
  data_paths = paths_track, width_ratio = 0.3, fill = c("#DC0000B2", "#00A087B2"),
  data_range = "auto"
)
```

`gghic()` is handy for adding all these features to the plot.

```{r}
x_5 |>
  dplyr::filter(
    center1 > 10000000 & center1 < 11000000 &
      center2 > 10000000 & center2 < 11000000
  ) |>
  gghic(
    draw_boundary = TRUE,

    ideogram = TRUE, genome = "hg19", highlight = TRUE, ideogram_fontsize = 7,
    ideogram_width_ratio = 0.08,
    annotation = TRUE, include_ncrna = FALSE, gtf_path = path_gtf,
    style = "arrow", maxgap = 100000, annotation_fontsize = 5,
    annotation_width_ratio = 0.05,

    track = TRUE, data_paths = paths_track, track_width_ratio = 0.5,
    track_fill = c("#DC0000B2", "#00A087B2"), data_range = "maximum",

    expand_xaxis = TRUE
  )
```

## Session Information

```{r}
sessionInfo()
```
