<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Add genomic signal tracks with direct data input (a second version) — geom_track2 • gghic</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Add genomic signal tracks with direct data input (a second version) — geom_track2"><meta name="description" content="Another version of geom_track() that accepts pre-loaded genomic signal
data as a data frame or tibble with explicit aesthetic mappings. Displays
continuous genomic signals (ChIP-seq, ATAC-seq, RNA-seq, etc.) as filled area
tracks below Hi-C heatmaps. Unlike geom_track(), this function uses
standard ggplot2 aesthetic mappings, making it more flexible for custom
styling and integration with ggplot2 scales and themes."><meta property="og:description" content="Another version of geom_track() that accepts pre-loaded genomic signal
data as a data frame or tibble with explicit aesthetic mappings. Displays
continuous genomic signals (ChIP-seq, ATAC-seq, RNA-seq, etc.) as filled area
tracks below Hi-C heatmaps. Unlike geom_track(), this function uses
standard ggplot2 aesthetic mappings, making it more flexible for custom
styling and integration with ggplot2 scales and themes."><meta property="og:image" content="https://jasonwong-lab.github.io/gghic/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gghic</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.2.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/gghic.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/depth.html">Resolution Depth Analysis for Hi-C/-like Data</a></li>
    <li><a class="dropdown-item" href="../articles/hypergraph.html">Multi-way Contact Analysis with Hypergraphs</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/jasonwong-lab/gghic/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Add genomic signal tracks with direct data input (a second version)</h1>
      <small class="dont-index">Source: <a href="https://github.com/jasonwong-lab/gghic/blob/master/R/geom-track2.R" class="external-link"><code>R/geom-track2.R</code></a></small>
      <div class="d-none name"><code>geom_track2.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Another version of <code><a href="geom_track.html">geom_track()</a></code> that accepts pre-loaded genomic signal
data as a data frame or tibble with explicit aesthetic mappings. Displays
continuous genomic signals (ChIP-seq, ATAC-seq, RNA-seq, etc.) as filled area
tracks below Hi-C heatmaps. Unlike <code><a href="geom_track.html">geom_track()</a></code>, this function uses
standard ggplot2 aesthetic mappings, making it more flexible for custom
styling and integration with ggplot2 scales and themes.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">geom_track2</span><span class="op">(</span></span>
<span>  mapping <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  data <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stat <span class="op">=</span> <span class="va">StatTrack2</span>,</span>
<span>  position <span class="op">=</span> <span class="st">"identity"</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show.legend <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  inherit.aes <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  width_ratio <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">20</span>,</span>
<span>  spacing_ratio <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  data_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"auto"</span>, <span class="st">"maximum"</span><span class="op">)</span>,</span>
<span>  rasterize <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  dpi <span class="op">=</span> <span class="fl">300</span>,</span>
<span>  dev <span class="op">=</span> <span class="st">"cairo"</span>,</span>
<span>  scale <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  draw_boundary <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  boundary_colour <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>  linetype <span class="op">=</span> <span class="st">"dashed"</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-mapping">mapping<a class="anchor" aria-label="anchor" href="#arg-mapping"></a></dt>
<dd><p>Set of aesthetic mappings created by <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p></dd>


<dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>The data to be displayed in this layer. There are three
options:</p>
<p>If <code>NULL</code>, the default, the data is inherited from the plot
data as specified in the call to <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot()</a></code>.</p>
<p>A <code>data.frame</code>, or other object, will override the plot
data. All objects will be fortified to produce a data frame. See
<code><a href="https://ggplot2.tidyverse.org/reference/fortify.html" class="external-link">fortify()</a></code> for which variables will be created.</p>
<p>A <code>function</code> will be called with a single argument,
the plot data. The return value must be a <code>data.frame</code>, and
will be used as the layer data. A <code>function</code> can be created
from a <code>formula</code> (e.g. <code>~ head(.x, 10)</code>).</p></dd>


<dt id="arg-stat">stat<a class="anchor" aria-label="anchor" href="#arg-stat"></a></dt>
<dd><p>The statistical transformation to use on the data for this layer.
When using a <code>geom_*()</code> function to construct a layer, the <code>stat</code>
argument can be used to override the default coupling between geoms and
stats. The <code>stat</code> argument accepts the following:</p><ul><li><p>A <code>Stat</code> ggproto subclass, for example <code>StatCount</code>.</p></li>
<li><p>A string naming the stat. To give the stat as a string, strip the
function name of the <code>stat_</code> prefix. For example, to use <code><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">stat_count()</a></code>,
give the stat as <code>"count"</code>.</p></li>
<li><p>For more information and other ways to specify the stat, see the
<a href="https://ggplot2.tidyverse.org/reference/layer_stats.html" class="external-link">layer stat</a> documentation.</p></li>
</ul></dd>


<dt id="arg-position">position<a class="anchor" aria-label="anchor" href="#arg-position"></a></dt>
<dd><p>A position adjustment to use on the data for this layer. This
can be used in various ways, including to prevent overplotting and
improving the display. The <code>position</code> argument accepts the following:</p><ul><li><p>The result of calling a position function, such as <code><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter()</a></code>.
This method allows for passing extra arguments to the position.</p></li>
<li><p>A string naming the position adjustment. To give the position as a
string, strip the function name of the <code>position_</code> prefix. For example,
to use <code><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter()</a></code>, give the position as <code>"jitter"</code>.</p></li>
<li><p>For more information and other ways to specify the position, see the
<a href="https://ggplot2.tidyverse.org/reference/layer_positions.html" class="external-link">layer position</a> documentation.</p></li>
</ul></dd>


<dt id="arg-na-rm">na.rm<a class="anchor" aria-label="anchor" href="#arg-na-rm"></a></dt>
<dd><p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p></dd>


<dt id="arg-show-legend">show.legend<a class="anchor" aria-label="anchor" href="#arg-show-legend"></a></dt>
<dd><p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display. To include legend keys for all levels, even
when no data exists, use <code>TRUE</code>.  If <code>NA</code>, all levels are shown in legend,
but unobserved levels are omitted.</p></dd>


<dt id="arg-inherit-aes">inherit.aes<a class="anchor" aria-label="anchor" href="#arg-inherit-aes"></a></dt>
<dd><p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="https://ggplot2.tidyverse.org/reference/annotation_borders.html" class="external-link">annotation_borders()</a></code>.</p></dd>


<dt id="arg-width-ratio">width_ratio<a class="anchor" aria-label="anchor" href="#arg-width-ratio"></a></dt>
<dd><p>Numeric value controlling the height of each track relative
to the Hi-C plot height. Smaller values create shorter tracks. Default is
<code>1/20</code> (5% of Hi-C plot height).</p></dd>


<dt id="arg-spacing-ratio">spacing_ratio<a class="anchor" aria-label="anchor" href="#arg-spacing-ratio"></a></dt>
<dd><p>Numeric value controlling the vertical spacing between
tracks as a proportion of track height. Default is <code>0.5</code> (50% of track height).</p></dd>


<dt id="arg-data-range">data_range<a class="anchor" aria-label="anchor" href="#arg-data-range"></a></dt>
<dd><p>Character string or numeric value(s) controlling y-axis scaling:</p><ul><li><p><code>"auto"</code>: Each track scaled independently to its own maximum value</p></li>
<li><p><code>"maximum"</code>: All tracks share the same y-axis scale (global maximum)</p></li>
<li><p>Numeric value: Fixed maximum for all tracks (e.g., <code>100</code>)</p></li>
<li><p>Numeric vector: Individual maximum for each track (length must match number of tracks)
Default is <code>"auto"</code>.</p></li>
</ul></dd>


<dt id="arg-rasterize">rasterize<a class="anchor" aria-label="anchor" href="#arg-rasterize"></a></dt>
<dd><p>Logical indicating whether to rasterize the track polygons for
faster rendering and smaller file sizes. Recommended for high-resolution data.
Default is <code>TRUE</code>.</p></dd>


<dt id="arg-dpi">dpi<a class="anchor" aria-label="anchor" href="#arg-dpi"></a></dt>
<dd><p>Numeric value specifying the resolution (dots per inch) for rasterized
tracks. Higher values increase quality but also file size. Default is <code>300</code>.</p></dd>


<dt id="arg-dev">dev<a class="anchor" aria-label="anchor" href="#arg-dev"></a></dt>
<dd><p>Character string specifying the graphics device for rasterization.
Options include <code>"cairo"</code>, <code>"ragg"</code>, or other devices. Default is <code>"cairo"</code>.</p></dd>


<dt id="arg-scale">scale<a class="anchor" aria-label="anchor" href="#arg-scale"></a></dt>
<dd><p>Numeric scaling factor for rasterized output. Values &gt; 1 increase
resolution. Default is <code>1</code>.</p></dd>


<dt id="arg-draw-boundary">draw_boundary<a class="anchor" aria-label="anchor" href="#arg-draw-boundary"></a></dt>
<dd><p>Logical indicating whether to draw vertical boundary lines
between chromosomes in multi-chromosome displays. Default is <code>TRUE</code>.</p></dd>


<dt id="arg-boundary-colour">boundary_colour<a class="anchor" aria-label="anchor" href="#arg-boundary-colour"></a></dt>
<dd><p>Character string specifying the color of chromosome
boundary lines. Default is <code>"black"</code>.</p></dd>


<dt id="arg-linetype">linetype<a class="anchor" aria-label="anchor" href="#arg-linetype"></a></dt>
<dd><p>Character string or integer specifying the line type for
chromosome boundaries. Default is <code>"dashed"</code>.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional parameters (currently ignored).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A ggplot2 layer object that can be added to a gghic plot.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>

<div class="section">
<h3 id="required-aesthetics">Required Aesthetics<a class="anchor" aria-label="anchor" href="#required-aesthetics"></a></h3>


<p>This geom requires the following aesthetics to be mapped:</p><ul><li><p><code>seqnames</code>: Chromosome name for each genomic bin</p></li>
<li><p><code>start</code>: Start position of each bin</p></li>
<li><p><code>end</code>: End position of each bin</p></li>
<li><p><code>score</code>: Signal intensity value (numeric)</p></li>
<li><p><code>name</code>: Track identifier/label (groups bins into tracks)</p></li>
</ul></div>

<div class="section">
<h3 id="comparison-with-geom-track-">Comparison with geom_track()<a class="anchor" aria-label="anchor" href="#comparison-with-geom-track-"></a></h3>
<table class="table table"><tr><td>Feature</td><td>geom_track()</td><td>geom_track2()</td></tr><tr><td>Input</td><td>File paths or GRanges list</td><td>Data frame with aesthetics</td></tr><tr><td>Flexibility</td><td>Simple, automatic</td><td>Full ggplot2 integration</td></tr><tr><td>Color control</td><td><code>fill</code> parameter</td><td><code>aes(fill = ...)</code> + scales</td></tr><tr><td>Data loading</td><td>Automatic from files</td><td>Manual pre-loading</td></tr><tr><td>Best for</td><td>Quick visualization</td><td>Custom styling &amp; scales</td></tr></table></div>

<div class="section">
<h3 id="input-data-format">Input Data Format<a class="anchor" aria-label="anchor" href="#input-data-format"></a></h3>


<p>Data should be a data frame or tibble where each row represents one genomic bin:</p>
<p></p><div class="sourceCode"><pre><code><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>seqnames  start    end  score      name</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>chr1      <span class="dv">10000</span>  <span class="dv">10100</span>   <span class="fl">5.2</span>    <span class="st">"ChIP-seq"</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>chr1      <span class="dv">10100</span>  <span class="dv">10200</span>   <span class="fl">6.1</span>    <span class="st">"ChIP-seq"</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>chr1      <span class="dv">10200</span>  <span class="dv">10300</span>   <span class="fl">4.8</span>    <span class="st">"ChIP-seq"</span></span></code></pre><p></p></div>
</div>

<div class="section">
<h3 id="track-visualization">Track Visualization<a class="anchor" aria-label="anchor" href="#track-visualization"></a></h3>


<p>Each track displays:</p><ul><li><p><strong>Signal area</strong>: Filled polygon showing signal intensity</p></li>
<li><p><strong>Y-axis</strong>: Left side with minimum (0) and maximum value labels</p></li>
<li><p><strong>Track label</strong>: Name (from <code>name</code> aesthetic) displayed on the left</p></li>
<li><p><strong>Baseline</strong>: Zero line at bottom of each track</p></li>
</ul></div>

<div class="section">
<h3 id="color-and-fill-aesthetics">Color and Fill Aesthetics<a class="anchor" aria-label="anchor" href="#color-and-fill-aesthetics"></a></h3>


<p>This function fully integrates with ggplot2's aesthetic system:</p><ul><li><p>Map <code>fill</code> aesthetic to track names for automatic coloring</p></li>
<li><p>Use <code><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual()</a></code>, <code><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer()</a></code>, etc. for custom colors</p></li>
<li><p>Supports all standard ggplot2 color specifications</p></li>
<li><p>Can combine with other aesthetic mappings (alpha, etc.)</p></li>
</ul></div>

<div class="section">
<h3 id="y-axis-scaling">Y-Axis Scaling<a class="anchor" aria-label="anchor" href="#y-axis-scaling"></a></h3>


<p>Control track scaling via <code>data_range</code>:</p><ul><li><p><strong>Independent</strong> (<code>"auto"</code>): Compare patterns within each track</p></li>
<li><p><strong>Shared</strong> (<code>"maximum"</code>): Compare absolute signal between tracks</p></li>
<li><p><strong>Fixed</strong> (numeric): Consistent scaling across multiple plots</p></li>
</ul></div>

<div class="section">
<h3 id="performance">Performance<a class="anchor" aria-label="anchor" href="#performance"></a></h3>

<ul><li><p>Rasterization recommended for dense genomic data</p></li>
<li><p>Adjust <code>dpi</code> to balance quality and file size</p></li>
<li><p>Pre-filter data to displayed region for faster rendering</p></li>
</ul></div>

<div class="section">
<h3 id="multi-chromosome-support">Multi-Chromosome Support<a class="anchor" aria-label="anchor" href="#multi-chromosome-support"></a></h3>


<p>When data includes multiple chromosomes:</p><ul><li><p>Coordinates automatically adjusted for proper alignment</p></li>
<li><p>Vertical boundaries separate chromosomes (if <code>draw_boundary = TRUE</code>)</p></li>
<li><p>Works seamlessly with Hi-C multi-chromosome displays</p></li>
</ul></div>

    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index">
<ul><li><p><code><a href="geom_track.html">geom_track()</a></code> for file-based input with automatic data loading</p></li>
<li><p><code><a href="gghic.html">gghic()</a></code> for creating the base Hi-C plot</p></li>
<li><p><code><a href="geom_annotation.html">geom_annotation()</a></code> for gene annotation tracks</p></li>
<li><p><code><a href="geom_hic.html">geom_hic()</a></code> for the Hi-C heatmap layer</p></li>
</ul></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Load Hi-C data</span></span></span>
<span class="r-in"><span><span class="va">cc</span> <span class="op">&lt;-</span> <span class="fu"><a href="ChromatinContacts.html">ChromatinContacts</a></span><span class="op">(</span><span class="st">"path/to/cooler.cool"</span>, focus <span class="op">=</span> <span class="st">"chr4"</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu">import</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Load and prepare track data</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">track1</span> <span class="op">&lt;-</span> <span class="fu">import</span><span class="op">(</span><span class="st">"path/to/H3K27ac.bw"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">track1</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="st">"H3K27ac"</span></span></span>
<span class="r-in"><span><span class="va">track_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">track1</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Basic usage with aesthetic mappings</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="gghic.html">gghic</a></span><span class="op">(</span><span class="va">cc</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">geom_track2</span><span class="op">(</span></span></span>
<span class="r-in"><span>    data <span class="op">=</span> <span class="va">track_df</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>      seqnames <span class="op">=</span> <span class="va">seqnames</span>,</span></span>
<span class="r-in"><span>      start <span class="op">=</span> <span class="va">start</span>,</span></span>
<span class="r-in"><span>      end <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>      score <span class="op">=</span> <span class="va">score</span>,</span></span>
<span class="r-in"><span>      name <span class="op">=</span> <span class="va">name</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Multiple tracks with color mapping</span></span></span>
<span class="r-in"><span><span class="va">track2</span> <span class="op">&lt;-</span> <span class="fu">import</span><span class="op">(</span><span class="st">"path/to/ATAC.bw"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">track2</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="st">"ATAC-seq"</span></span></span>
<span class="r-in"><span><span class="va">tracks_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">track1</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">track2</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="gghic.html">gghic</a></span><span class="op">(</span><span class="va">cc</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">geom_track2</span><span class="op">(</span></span></span>
<span class="r-in"><span>    data <span class="op">=</span> <span class="va">tracks_df</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>      seqnames <span class="op">=</span> <span class="va">seqnames</span>,</span></span>
<span class="r-in"><span>      start <span class="op">=</span> <span class="va">start</span>,</span></span>
<span class="r-in"><span>      end <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>      score <span class="op">=</span> <span class="va">score</span>,</span></span>
<span class="r-in"><span>      name <span class="op">=</span> <span class="va">name</span>,</span></span>
<span class="r-in"><span>      fill <span class="op">=</span> <span class="va">name</span>  <span class="co"># Color by track name</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"H3K27ac"</span> <span class="op">=</span> <span class="st">"#E63946"</span>, <span class="st">"ATAC-seq"</span> <span class="op">=</span> <span class="st">"#457B9D"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Using scale_fill_brewer for automatic colors</span></span></span>
<span class="r-in"><span><span class="fu"><a href="gghic.html">gghic</a></span><span class="op">(</span><span class="va">cc</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">geom_track2</span><span class="op">(</span></span></span>
<span class="r-in"><span>    data <span class="op">=</span> <span class="va">tracks_df</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">seqnames</span>, start <span class="op">=</span> <span class="va">start</span>, end <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        score <span class="op">=</span> <span class="va">score</span>, name <span class="op">=</span> <span class="va">name</span>, fill <span class="op">=</span> <span class="va">name</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set2"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Shared y-axis scaling</span></span></span>
<span class="r-in"><span><span class="fu"><a href="gghic.html">gghic</a></span><span class="op">(</span><span class="va">cc</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">geom_track2</span><span class="op">(</span></span></span>
<span class="r-in"><span>    data <span class="op">=</span> <span class="va">tracks_df</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">seqnames</span>, start <span class="op">=</span> <span class="va">start</span>, end <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        score <span class="op">=</span> <span class="va">score</span>, name <span class="op">=</span> <span class="va">name</span>, fill <span class="op">=</span> <span class="va">name</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    data_range <span class="op">=</span> <span class="st">"maximum"</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Custom track dimensions</span></span></span>
<span class="r-in"><span><span class="fu"><a href="gghic.html">gghic</a></span><span class="op">(</span><span class="va">cc</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">geom_track2</span><span class="op">(</span></span></span>
<span class="r-in"><span>    data <span class="op">=</span> <span class="va">track_df</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">seqnames</span>, start <span class="op">=</span> <span class="va">start</span>, end <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        score <span class="op">=</span> <span class="va">score</span>, name <span class="op">=</span> <span class="va">name</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    width_ratio <span class="op">=</span> <span class="fl">1</span> <span class="op">/</span> <span class="fl">15</span>,    <span class="co"># Taller tracks</span></span></span>
<span class="r-in"><span>    spacing_ratio <span class="op">=</span> <span class="fl">1</span>         <span class="co"># More spacing</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Fixed y-axis maximum</span></span></span>
<span class="r-in"><span><span class="fu"><a href="gghic.html">gghic</a></span><span class="op">(</span><span class="va">cc</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">geom_track2</span><span class="op">(</span></span></span>
<span class="r-in"><span>    data <span class="op">=</span> <span class="va">track_df</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">seqnames</span>, start <span class="op">=</span> <span class="va">start</span>, end <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        score <span class="op">=</span> <span class="va">score</span>, name <span class="op">=</span> <span class="va">name</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    data_range <span class="op">=</span> <span class="fl">50</span>  <span class="co"># Fix max to 50</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Multiple tracks with different y-axis limits</span></span></span>
<span class="r-in"><span><span class="fu"><a href="gghic.html">gghic</a></span><span class="op">(</span><span class="va">cc</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">geom_track2</span><span class="op">(</span></span></span>
<span class="r-in"><span>    data <span class="op">=</span> <span class="va">tracks_df</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">seqnames</span>, start <span class="op">=</span> <span class="va">start</span>, end <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        score <span class="op">=</span> <span class="va">score</span>, name <span class="op">=</span> <span class="va">name</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    data_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">50</span><span class="op">)</span>  <span class="co"># Different max for each track</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># High-resolution rasterization</span></span></span>
<span class="r-in"><span><span class="fu"><a href="gghic.html">gghic</a></span><span class="op">(</span><span class="va">cc</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">geom_track2</span><span class="op">(</span></span></span>
<span class="r-in"><span>    data <span class="op">=</span> <span class="va">track_df</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">seqnames</span>, start <span class="op">=</span> <span class="va">start</span>, end <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        score <span class="op">=</span> <span class="va">score</span>, name <span class="op">=</span> <span class="va">name</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    rasterize <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    dpi <span class="op">=</span> <span class="fl">600</span>,</span></span>
<span class="r-in"><span>    dev <span class="op">=</span> <span class="st">"ragg"</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Combine with other geoms</span></span></span>
<span class="r-in"><span><span class="fu"><a href="gghic.html">gghic</a></span><span class="op">(</span><span class="va">cc</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">geom_track2</span><span class="op">(</span></span></span>
<span class="r-in"><span>    data <span class="op">=</span> <span class="va">track_df</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">seqnames</span>, start <span class="op">=</span> <span class="va">start</span>, end <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        score <span class="op">=</span> <span class="va">score</span>, name <span class="op">=</span> <span class="va">name</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    fill <span class="op">=</span> <span class="st">"darkblue"</span></span></span>
<span class="r-in"><span>  <span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="geom_loop.html">geom_loop</a></span><span class="op">(</span><span class="st">"path/to/loops.bedpe"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Multi-chromosome display</span></span></span>
<span class="r-in"><span><span class="va">cc_multi</span> <span class="op">&lt;-</span> <span class="fu"><a href="ChromatinContacts.html">ChromatinContacts</a></span><span class="op">(</span><span class="st">"path/to/cooler.cool"</span>,</span></span>
<span class="r-in"><span>                              focus <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr4"</span>, <span class="st">"chr8"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu">import</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="gghic.html">gghic</a></span><span class="op">(</span><span class="va">cc_multi</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">geom_track2</span><span class="op">(</span></span></span>
<span class="r-in"><span>    data <span class="op">=</span> <span class="va">tracks_df</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="va">seqnames</span>, start <span class="op">=</span> <span class="va">start</span>, end <span class="op">=</span> <span class="va">end</span>,</span></span>
<span class="r-in"><span>        score <span class="op">=</span> <span class="va">score</span>, name <span class="op">=</span> <span class="va">name</span>, fill <span class="op">=</span> <span class="va">name</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    draw_boundary <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Minghao Jiang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

