<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>gghic — gghic • gghic</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="gghic — gghic"><meta name="description" content="A ggplot2 wrapper of geom_hic(), geom_ideogram(),
geom_annotation(), geom_track(), geom_tad(), and geom_loop()
for easy visualisation of Hi-C/-like data."><meta property="og:description" content="A ggplot2 wrapper of geom_hic(), geom_ideogram(),
geom_annotation(), geom_track(), geom_tad(), and geom_loop()
for easy visualisation of Hi-C/-like data."><meta property="og:image" content="https://jasonwong-lab.github.io/gghic/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gghic</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/gghic.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/jasonwong-lab/gghic/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>gghic</h1>
      <small class="dont-index">Source: <a href="https://github.com/jasonwong-lab/gghic/blob/master/R/gghic.R" class="external-link"><code>R/gghic.R</code></a></small>
      <div class="d-none name"><code>gghic.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>A ggplot2 wrapper of <code><a href="geom_hic.html">geom_hic()</a></code>, <code><a href="geom_ideogram.html">geom_ideogram()</a></code>,
<code><a href="geom_annotation.html">geom_annotation()</a></code>, <code><a href="geom_track.html">geom_track()</a></code>, <code><a href="geom_tad.html">geom_tad()</a></code>, and <code><a href="geom_loop.html">geom_loop()</a></code>
for easy visualisation of Hi-C/-like data.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">gghic</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  score_column <span class="op">=</span> <span class="st">"balanced"</span>,</span>
<span>  scale_method <span class="op">=</span> <span class="va">log10</span>,</span>
<span>  ideogram <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ideogram_width_ratio <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">30</span>,</span>
<span>  ideogram_fontsize <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  ideogram_colour <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>  ideogram_fill <span class="op">=</span> <span class="st">"#FFE3E680"</span>,</span>
<span>  annotation <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  annotation_width_ratio <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">50</span>,</span>
<span>  annotation_spacing_ratio <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">3</span>,</span>
<span>  annotation_fontsize <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  annotation_colour <span class="op">=</span> <span class="st">"#48CFCB"</span>,</span>
<span>  annotation_fill <span class="op">=</span> <span class="st">"#48CFCB"</span>,</span>
<span>  track <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  track_width_ratio <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">20</span>,</span>
<span>  track_spacing_ratio <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">2</span>,</span>
<span>  track_fill <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>  track_fontsize <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  tad <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  tad_is_0based <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  tad_colour <span class="op">=</span> <span class="st">"grey"</span>,</span>
<span>  loop <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  loop_is_0based <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  loop_colour <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>  loop_fill <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  expand_xaxis <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  expand_left <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  expand_right <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>Either a HiCExperiment object, or a GInteractions object, or a
tibble/data.frame.</p></dd>


<dt id="arg-score-column">score_column<a class="anchor" aria-label="anchor" href="#arg-score-column"></a></dt>
<dd><p>The column name of which the score is calculated.
Default is <code>"balanced"</code>.</p></dd>


<dt id="arg-scale-method">scale_method<a class="anchor" aria-label="anchor" href="#arg-scale-method"></a></dt>
<dd><p>The function to scale the score. Default is <code>log10</code>.</p></dd>


<dt id="arg-ideogram">ideogram<a class="anchor" aria-label="anchor" href="#arg-ideogram"></a></dt>
<dd><p>Whether to add ideogram or not. Default is <code>FALSE</code>.</p></dd>


<dt id="arg-ideogram-width-ratio">ideogram_width_ratio<a class="anchor" aria-label="anchor" href="#arg-ideogram-width-ratio"></a></dt>
<dd><p>The width ratio of the ideogram.
Default is <code>1/30</code>.</p></dd>


<dt id="arg-ideogram-fontsize">ideogram_fontsize<a class="anchor" aria-label="anchor" href="#arg-ideogram-fontsize"></a></dt>
<dd><p>The font size of the ideogram. Default is <code>10</code>.</p></dd>


<dt id="arg-ideogram-colour">ideogram_colour<a class="anchor" aria-label="anchor" href="#arg-ideogram-colour"></a></dt>
<dd><p>The color of the ideogram. Default is <code>"red"</code>.</p></dd>


<dt id="arg-ideogram-fill">ideogram_fill<a class="anchor" aria-label="anchor" href="#arg-ideogram-fill"></a></dt>
<dd><p>The fill color of the highlighted region on the
ideogram. Default is <code>"#FFE3E680"</code>.</p></dd>


<dt id="arg-annotation">annotation<a class="anchor" aria-label="anchor" href="#arg-annotation"></a></dt>
<dd><p>Whether to add annotation or not. Default is <code>FALSE</code>.</p></dd>


<dt id="arg-annotation-width-ratio">annotation_width_ratio<a class="anchor" aria-label="anchor" href="#arg-annotation-width-ratio"></a></dt>
<dd><p>The width ratio of the annotation.
Default is <code>1/50</code>.</p></dd>


<dt id="arg-annotation-spacing-ratio">annotation_spacing_ratio<a class="anchor" aria-label="anchor" href="#arg-annotation-spacing-ratio"></a></dt>
<dd><p>The spacing ratio of the annotation.
Default is <code>1/3</code>.</p></dd>


<dt id="arg-annotation-fontsize">annotation_fontsize<a class="anchor" aria-label="anchor" href="#arg-annotation-fontsize"></a></dt>
<dd><p>The font size of the annotation. Default is <code>10</code>.</p></dd>


<dt id="arg-annotation-colour">annotation_colour<a class="anchor" aria-label="anchor" href="#arg-annotation-colour"></a></dt>
<dd><p>The color of the annotation. Default is <code>"#48CFCB"</code>.</p></dd>


<dt id="arg-annotation-fill">annotation_fill<a class="anchor" aria-label="anchor" href="#arg-annotation-fill"></a></dt>
<dd><p>The fill color of the annotation.
Default is <code>"#48CFCB"</code>.</p></dd>


<dt id="arg-track">track<a class="anchor" aria-label="anchor" href="#arg-track"></a></dt>
<dd><p>Whether to add track or not. Default is <code>FALSE</code>.</p></dd>


<dt id="arg-track-width-ratio">track_width_ratio<a class="anchor" aria-label="anchor" href="#arg-track-width-ratio"></a></dt>
<dd><p>The width ratio of the track. Default is <code>1/20</code>.</p></dd>


<dt id="arg-track-spacing-ratio">track_spacing_ratio<a class="anchor" aria-label="anchor" href="#arg-track-spacing-ratio"></a></dt>
<dd><p>The spacing ratio of the track. Default is <code>1/2</code>.</p></dd>


<dt id="arg-track-fill">track_fill<a class="anchor" aria-label="anchor" href="#arg-track-fill"></a></dt>
<dd><p>The fill color of the track. Default is <code>"black"</code>.</p></dd>


<dt id="arg-track-fontsize">track_fontsize<a class="anchor" aria-label="anchor" href="#arg-track-fontsize"></a></dt>
<dd><p>The font size of the track. Default is <code>5</code>.</p></dd>


<dt id="arg-tad">tad<a class="anchor" aria-label="anchor" href="#arg-tad"></a></dt>
<dd><p>Whether to add TAD or not. Default is <code>FALSE</code>.</p></dd>


<dt id="arg-tad-is-based">tad_is_0based<a class="anchor" aria-label="anchor" href="#arg-tad-is-based"></a></dt>
<dd><p>Whether the TAD coordinates are 0-based or not.
Default is <code>FALSE</code>.</p></dd>


<dt id="arg-tad-colour">tad_colour<a class="anchor" aria-label="anchor" href="#arg-tad-colour"></a></dt>
<dd><p>The color of the TAD. Default is <code>"grey"</code>.</p></dd>


<dt id="arg-loop">loop<a class="anchor" aria-label="anchor" href="#arg-loop"></a></dt>
<dd><p>Whether to add loop or not. Default is <code>FALSE</code>.</p></dd>


<dt id="arg-loop-is-based">loop_is_0based<a class="anchor" aria-label="anchor" href="#arg-loop-is-based"></a></dt>
<dd><p>Whether the loop coordinates are 0-based or not.
Default is <code>FALSE</code>.</p></dd>


<dt id="arg-loop-colour">loop_colour<a class="anchor" aria-label="anchor" href="#arg-loop-colour"></a></dt>
<dd><p>The color of the loop. Default is <code>"black"</code>.</p></dd>


<dt id="arg-loop-fill">loop_fill<a class="anchor" aria-label="anchor" href="#arg-loop-fill"></a></dt>
<dd><p>The fill color of the loop. Default is <code>NA</code>.</p></dd>


<dt id="arg-expand-xaxis">expand_xaxis<a class="anchor" aria-label="anchor" href="#arg-expand-xaxis"></a></dt>
<dd><p>Whether to expand the x-axis or not. Default is <code>FALSE</code>.</p></dd>


<dt id="arg-expand-left">expand_left<a class="anchor" aria-label="anchor" href="#arg-expand-left"></a></dt>
<dd><p>The left expansion of the x-axis. Default is <code>NULL</code>.</p></dd>


<dt id="arg-expand-right">expand_right<a class="anchor" aria-label="anchor" href="#arg-expand-right"></a></dt>
<dd><p>The right expansion of the x-axis. Default is <code>NULL</code>.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Arguments passed on to <code><a href="geom_tad.html">geom_tad</a></code>, <code><a href="geom_loop.html">geom_loop</a></code>, <code><a href="geom_hic.html">geom_hic</a></code>, <code><a href="geom_ideogram.html">geom_ideogram</a></code>, <code><a href="geom_annotation.html">geom_annotation</a></code>, <code><a href="geom_track.html">geom_track</a></code>, <code><a href="geom_tad.html">geom_tad</a></code>, <code><a href="geom_loop.html">geom_loop</a></code></p><dl><dt><code>tad_path</code></dt>
<dd><p>A path to the TAD file. Default is <code>NULL</code>.</p></dd>

    <dt><code>tad_gis</code></dt>
<dd><p>An InteractionSet object of TADs. Default is <code>NULL</code>.</p></dd>

    <dt><code>is_0based</code></dt>
<dd><p>Whether the TAD file is 0-based or not. Default is <code>FALSE</code>.</p></dd>

    <dt><code>colour</code></dt>
<dd><p>The color of the TADs. Default is <code>grey</code>.</p></dd>

    <dt><code>mapping</code></dt>
<dd><p>Set of aesthetic mappings created by <code><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes()</a></code>. If specified and
<code>inherit.aes = TRUE</code> (the default), it is combined with the default mapping
at the top level of the plot. You must supply <code>mapping</code> if there is no plot
mapping.</p></dd>

    <dt><code>stat</code></dt>
<dd><p>The statistical transformation to use on the data for this layer.
When using a <code>geom_*()</code> function to construct a layer, the <code>stat</code>
argument can be used the override the default coupling between geoms and
stats. The <code>stat</code> argument accepts the following:</p><ul><li><p>A <code>Stat</code> ggproto subclass, for example <code>StatCount</code>.</p></li>
<li><p>A string naming the stat. To give the stat as a string, strip the
function name of the <code>stat_</code> prefix. For example, to use <code><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">stat_count()</a></code>,
give the stat as <code>"count"</code>.</p></li>
<li><p>For more information and other ways to specify the stat, see the
<a href="https://ggplot2.tidyverse.org/reference/layer_stats.html" class="external-link">layer stat</a> documentation.</p></li>
</ul></dd>

    <dt><code>position</code></dt>
<dd><p>A position adjustment to use on the data for this layer. This
can be used in various ways, including to prevent overplotting and
improving the display. The <code>position</code> argument accepts the following:</p><ul><li><p>The result of calling a position function, such as <code><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter()</a></code>.
This method allows for passing extra arguments to the position.</p></li>
<li><p>A string naming the position adjustment. To give the position as a
string, strip the function name of the <code>position_</code> prefix. For example,
to use <code><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter()</a></code>, give the position as <code>"jitter"</code>.</p></li>
<li><p>For more information and other ways to specify the position, see the
<a href="https://ggplot2.tidyverse.org/reference/layer_positions.html" class="external-link">layer position</a> documentation.</p></li>
</ul></dd>

    <dt><code>na.rm</code></dt>
<dd><p>If <code>FALSE</code>, the default, missing values are removed with
a warning. If <code>TRUE</code>, missing values are silently removed.</p></dd>

    <dt><code>show.legend</code></dt>
<dd><p>logical. Should this layer be included in the legends?
<code>NA</code>, the default, includes if any aesthetics are mapped.
<code>FALSE</code> never includes, and <code>TRUE</code> always includes.
It can also be a named logical vector to finely select the aesthetics to
display.</p></dd>

    <dt><code>inherit.aes</code></dt>
<dd><p>If <code>FALSE</code>, overrides the default aesthetics,
rather than combining with them. This is most useful for helper functions
that define both data and aesthetics and shouldn't inherit behaviour from
the default plot specification, e.g. <code><a href="https://ggplot2.tidyverse.org/reference/borders.html" class="external-link">borders()</a></code>.</p></dd>

    <dt><code>loop_path</code></dt>
<dd><p>A path to the loop file. Default is <code>NULL</code>.</p></dd>

    <dt><code>loop_gis</code></dt>
<dd><p>An InteractionSet object of loops. Default is <code>NULL</code>.</p></dd>

    <dt><code>shape</code></dt>
<dd><p>The shape of the loops. Default is <code>21</code>.</p></dd>

    <dt><code>fill</code></dt>
<dd><p>The fill color of the loops. Default is <code>NA</code>.</p></dd>

    <dt><code>size</code></dt>
<dd><p>The size of the loops. Default is <code>NULL</code>.
If <code>NULL</code>, the size is <code>grid::unit(1 / 80, "native")</code>.</p></dd>

    <dt><code>linewidth</code></dt>
<dd><p>The line width of the loops. Default is <code>1</code>.</p></dd>

    <dt><code>rasterize</code></dt>
<dd><p>Whether to rasterize the plot or not. Default is <code>FALSE</code>.</p></dd>

    <dt><code>dpi</code></dt>
<dd><p>The resolution of the rasterised plot. Default is <code>300</code>.</p></dd>

    <dt><code>dev</code></dt>
<dd><p>The device to rasterise the plot. Default is <code>"cairo"</code>.</p></dd>

    <dt><code>scale</code></dt>
<dd><p>The scale of the rasterised plot. Default is <code>1</code>.</p></dd>

    <dt><code>draw_boundary</code></dt>
<dd><p>Whether to draw the boundary line or not when plotting
multiple chromosomes. Default is <code>TRUE</code>.</p></dd>

    <dt><code>boundary_colour</code></dt>
<dd><p>The color of the boundary line. Default is <code>"black"</code>.</p></dd>

    <dt><code>linetype</code></dt>
<dd><p>The line type of the boundary line. Default is <code>"dashed"</code>.</p></dd>

    <dt><code>genome</code></dt>
<dd><p>The genome name. Default is <code>"hg19"</code>.</p></dd>

    <dt><code>chrom_prefix</code></dt>
<dd><p>Whether the input data has chromosome names
with prefix 'chr' or not. Default is <code>TRUE</code>.</p></dd>

    <dt><code>highlight</code></dt>
<dd><p>Whether to highlight the boundary of the chromosome.
Default is <code>TRUE</code>.</p></dd>

    <dt><code>length_ratio</code></dt>
<dd><p>The ratio of the length of each chromosome ideogram
relative to the width of the Hi-C plot. Default is <code>0.8</code>.</p></dd>

    <dt><code>txdb</code></dt>
<dd><p>The TxDb object. Default is <code>NULL</code>.</p></dd>

    <dt><code>tx2gene</code></dt>
<dd><p>An optional data frame or tibble that maps transcript
information to gene information. It should include the following columns:</p><ul><li><p>chrom: Chromosome number or name.</p></li>
<li><p>gene_id: Entrez gene ID.</p></li>
<li><p>gene_symbol: Common symbol or name of the gene.</p></li>
<li><p>tx_id: Entrez transcript ID.</p></li>
<li><p>tx_name: Name of the transcript.</p></li>
<li><p>gene_type: Type or classification of the gene.</p></li>
<li><p>tx_type: Type or classification of the transcript.</p></li>
</ul></dd>

    <dt><code>gtf_path</code></dt>
<dd><p>The path to the GTF file, which is used to generate <code>txdb</code>
and <code>tx2gene</code>. Generated files are saved in the cache directory.
Default is <code>NULL</code>.</p></dd>

    <dt><code>maxgap</code></dt>
<dd><p>The maximum gap between genes to be drawn in the same line.
Default is <code>-1</code>.</p></dd>

    <dt><code>include_ncrna</code></dt>
<dd><p>Whether to include ncRNA or not. Default is <code>TRUE</code>.</p></dd>

    <dt><code>style</code></dt>
<dd><p>The style of the gene model track, which can be <code>"basic"</code>
or <code>"arrow"</code>. Default is <code>"basic"</code>.</p></dd>

    <dt><code>gene_symbols</code></dt>
<dd><p>A character vector of gene symbols to be included only.
Default is <code>NULL</code>.</p></dd>

    <dt><code>fontsize</code></dt>
<dd><p>The font size of the gene symbols. Default is <code>10</code>.</p></dd>

    <dt><code>data_paths</code></dt>
<dd><p>The paths to the sequencing data files. Default is <code>NULL</code>.</p></dd>

    <dt><code>data_granges</code></dt>
<dd><p>The GRanges object of the sequencing data.
Default is <code>NULL</code>.</p></dd>

    <dt><code>data_range</code></dt>
<dd><p>The range of the x axis. It can be <code>"auto"</code>, <code>"maximum"</code>,
or a number (vector). Default is <code>"auto"</code>.</p></dd>


</dl></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A ggplot object.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jasonwong-lab/gghic" class="external-link">gghic</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/js2264/HiCExperiment" class="external-link">HiCExperiment</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">InteractionSet</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://scales.r-lib.org" class="external-link">scales</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://scales.r-lib.org" class="external-link">scales</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="download_example_files.html">download_example_files</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dir_cache_gghic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rappdirs.r-lib.org/reference/user_cache_dir.html" class="external-link">user_cache_dir</a></span><span class="op">(</span>appname <span class="op">=</span> <span class="st">"gghic"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">hic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://glue.tidyverse.org/reference/glue.html" class="external-link">glue</a></span><span class="op">(</span><span class="st">"{dir_cache_gghic}/chr4_11-5kb.cool"</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/pkg/HiCExperiment/man/CoolFile-class.html" class="external-link">CoolFile</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/pkg/BiocIO/man/IO.html" class="external-link">import</a></span><span class="op">(</span><span class="va">cf</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">gis</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/interaction-accessors.html" class="external-link">interactions</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gis</span><span class="op">$</span><span class="va">score</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">gis</span><span class="op">$</span><span class="va">balanced</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">gis</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">score</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/distances.html" class="external-link">pairdist</a></span><span class="op">(</span><span class="va">gis</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span> <span class="op">&amp;</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/distances.html" class="external-link">pairdist</a></span><span class="op">(</span><span class="va">gis</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="va">scores</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.infinite</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">x</span><span class="op">$</span><span class="va">score</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://scales.r-lib.org/reference/oob.html" class="external-link">oob_squish</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">score</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">x</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">center1</span> <span class="op">&gt;</span> <span class="fl">10000000</span> <span class="op">&amp;</span> <span class="va">center1</span> <span class="op">&lt;</span> <span class="fl">11000000</span> <span class="op">&amp;</span></span></span>
<span class="r-in"><span>      <span class="va">center2</span> <span class="op">&gt;</span> <span class="fl">10000000</span> <span class="op">&amp;</span> <span class="va">center2</span> <span class="op">&lt;</span> <span class="fl">11000000</span></span></span>
<span class="r-in"><span>  <span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu">gghic</span><span class="op">(</span></span></span>
<span class="r-in"><span>    draw_boundary <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    ideogram <span class="op">=</span> <span class="cn">TRUE</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, highlight <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    ideogram_fontsize <span class="op">=</span> <span class="fl">7</span>, ideogram_width_ratio <span class="op">=</span> <span class="fl">0.08</span>,</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    annotation <span class="op">=</span> <span class="cn">TRUE</span>, include_ncrna <span class="op">=</span> <span class="cn">FALSE</span>, gtf_path <span class="op">=</span> <span class="va">path_gtf</span>,</span></span>
<span class="r-in"><span>    style <span class="op">=</span> <span class="st">"arrow"</span>, maxgap <span class="op">=</span> <span class="fl">100000</span>, annotation_fontsize <span class="op">=</span> <span class="fl">5</span>,</span></span>
<span class="r-in"><span>    annotation_width_ratio <span class="op">=</span> <span class="fl">0.05</span>,</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    expand_xaxis <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Minghao Jiang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer></div>





  </body></html>

