<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index • gghic</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><meta property="og:image" content="https://jasonwong-lab.github.io/gghic/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gghic</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.2.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/gghic.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/depth.html">Resolution Depth Analysis for Hi-C/-like Data</a></li>
    <li><a class="dropdown-item" href="../articles/hypergraph.html">Multi-way Contact Analysis with Hypergraphs</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/jasonwong-lab/gghic/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>



      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="ChromatinContacts-class.html">ChromatinContacts-class</a></code>

        </dt>
        <dd>ChromatinContacts S4 class</dd>

        <dt>

          <code><a href="ChromatinContacts.html">ChromatinContacts()</a></code>

        </dt>
        <dd>Create a ChromatinContacts object</dd>

        <dt>

          <code><a href="MultiWayContacts-class.html">MultiWayContacts()</a></code>

        </dt>
        <dd>MultiWayContacts S4 class</dd>

        <dt>

          <code><a href="as_tibble.html">as_tibble()</a></code>

        </dt>
        <dd>Convert ChromatinContacts to tibble</dd>

        <dt>

          <code><a href="build.html">build()</a></code>

        </dt>
        <dd>Build hypergraph from multi-way contacts</dd>

        <dt>

          <code><a href="concatemers2gis.html">concatemers2Gis()</a></code>

        </dt>
        <dd>Convert multi-way concatemer reads to pairwise Hi-C interactions</dd>

        <dt>

          <code><a href="features-set.html">`features&lt;-`()</a></code>

        </dt>
        <dd>Set genomic features in ChromatinContacts object</dd>

        <dt>

          <code><a href="features.html">features()</a></code>

        </dt>
        <dd>Get genomic features</dd>

        <dt>

          <code><a href="focus.html">focus()</a></code>

        </dt>
        <dd>Get focus regions</dd>

        <dt>

          <code><a href="geom_annotation.html">geom_annotation()</a></code>

        </dt>
        <dd>Add gene annotation tracks to Hi-C plots</dd>

        <dt>

          <code><a href="geom_concatemer.html">geom_concatemer()</a></code>

        </dt>
        <dd>Visualize multi-way contact concatemers below Hi-C map</dd>

        <dt>

          <code><a href="geom_concatemer2.html">geom_concatemer2()</a></code>

        </dt>
        <dd>Visualize concatemers (a second version)</dd>

        <dt>

          <code><a href="geom_hic.html">geom_hic()</a></code>

        </dt>
        <dd>geom_hic</dd>

        <dt>

          <code><a href="geom_hic_under.html">geom_hic_under()</a></code>

        </dt>
        <dd>Visualize inverted Hi-C heatmap below main contact map</dd>

        <dt>

          <code><a href="geom_hypergraph.html">geom_hypergraph()</a></code>

        </dt>
        <dd>Visualize Multi-way Contacts as Hypergraph</dd>

        <dt>

          <code><a href="geom_ideogram.html">geom_ideogram()</a></code>

        </dt>
        <dd>Add chromosome ideogram annotation to Hi-C plot</dd>

        <dt>

          <code><a href="geom_loop.html">geom_loop()</a></code>

        </dt>
        <dd>Visualize chromatin loops on Hi-C heatmap</dd>

        <dt>

          <code><a href="geom_loop2.html">geom_loop2()</a></code>

        </dt>
        <dd>Visualize chromatin loops (a second version)</dd>

        <dt>

          <code><a href="geom_tad.html">geom_tad()</a></code>

        </dt>
        <dd>Visualize TAD boundaries on Hi-C heatmap</dd>

        <dt>

          <code><a href="geom_tad2.html">geom_tad2()</a></code>

        </dt>
        <dd>Visualize TAD boundaries (a second version)</dd>

        <dt>

          <code><a href="geom_track.html">geom_track()</a></code>

        </dt>
        <dd>Add genomic signal tracks to Hi-C plots</dd>

        <dt>

          <code><a href="geom_track2.html">geom_track2()</a></code>

        </dt>
        <dd>Add genomic signal tracks with direct data input (a second version)</dd>

        <dt>

          <code><a href="getBreaksLabels.html">getBreaksLabels()</a></code>

        </dt>
        <dd>Generate axis breaks and labels for multi-chromosome plots</dd>

        <dt>

          <code><a href="gghic.html">gghic()</a></code>

        </dt>
        <dd>Create Hi-C visualization plot</dd>

        <dt>

          <code><a href="gghypergraph.html">gghypergraph()</a></code>

        </dt>
        <dd>Visualize multi-way chromatin contacts as hypergraph network</dd>

        <dt>

          <code><a href="hypergraphData.html">hypergraphData()</a></code>

        </dt>
        <dd>Get hypergraph data from MultiWayContacts object</dd>

        <dt>

          <code><a href="import-ChromatinContacts.html">import(<i>&lt;ChromatinContacts&gt;</i>,<i>&lt;ANY&gt;</i>,<i>&lt;ANY&gt;</i>)</a></code>

        </dt>
        <dd>Import chromatin contact data</dd>

        <dt>

          <code><a href="import-MultiWayContacts.html">import(<i>&lt;MultiWayContacts&gt;</i>,<i>&lt;ANY&gt;</i>,<i>&lt;ANY&gt;</i>)</a></code>

        </dt>
        <dd>Import multi-way contact data</dd>

        <dt>

          <code><a href="interactions-ChromatinContacts-method.html">interactions(<i>&lt;ChromatinContacts&gt;</i>)</a></code>

        </dt>
        <dd>Get interaction data</dd>

        <dt>

          <code><a href="plotCoverageCurve.html">plotCoverageCurve()</a></code>

        </dt>
        <dd>Plot Coverage Fraction by Resolution</dd>

        <dt>

          <code><a href="plotResolutionCoverage.html">plotResolutionCoverage()</a></code>

        </dt>
        <dd>Plot Resolution vs Coverage</dd>

        <dt>

          <code><a href="resolution-depth.html">calculateResolutionDepth()</a></code> <code><a href="resolution-depth.html">calculateGenomeCoverage()</a></code> <code><a href="resolution-depth.html">findOptimalResolution()</a></code> <code><a href="resolution-depth.html">readPairsCache()</a></code> <code><a href="resolution-depth.html">calcResDepthChunked()</a></code> <code><a href="resolution-depth.html">calcGenomeCovChunked()</a></code> <code><a href="resolution-depth.html">findOptResChunked()</a></code>

        </dt>
        <dd>Calculate Resolution and Depth for Pairs Data</dd>

        <dt>

          <code><a href="resolution.html">resolution()</a></code>

        </dt>
        <dd>Get resolution of ChromatinContacts object</dd>

        <dt>

          <code><a href="scaleData.html">scaleData()</a></code>

        </dt>
        <dd>Scale and transform Hi-C interaction data for visualization</dd>

        <dt>

          <code><a href="select.html">select()</a></code>

        </dt>
        <dd>Select top hyperedges for visualization</dd>

        <dt>

          <code><a href="seqinfo-ChromatinContacts-method.html">seqinfo(<i>&lt;ChromatinContacts&gt;</i>)</a></code>

        </dt>
        <dd>Get sequence information</dd>

        <dt>

          <code><a href="subset-methods.html">subsetByOverlaps(<i>&lt;ChromatinContacts&gt;</i>,<i>&lt;numeric&gt;</i>)</a></code> <code><a href="subset-methods.html">subsetByOverlaps(<i>&lt;ChromatinContacts&gt;</i>,<i>&lt;logical&gt;</i>)</a></code> <code><a href="subset-methods.html">subsetByOverlaps(<i>&lt;ChromatinContacts&gt;</i>,<i>&lt;GRanges&gt;</i>)</a></code> <code><a href="subset-methods.html">subsetByOverlaps(<i>&lt;ChromatinContacts&gt;</i>,<i>&lt;GInteractions&gt;</i>)</a></code> <code><a href="subset-methods.html">`[`(<i>&lt;ChromatinContacts&gt;</i>,<i>&lt;numeric&gt;</i>,<i>&lt;ANY&gt;</i>,<i>&lt;ANY&gt;</i>)</a></code> <code><a href="subset-methods.html">`[`(<i>&lt;ChromatinContacts&gt;</i>,<i>&lt;logical&gt;</i>,<i>&lt;ANY&gt;</i>,<i>&lt;ANY&gt;</i>)</a></code> <code><a href="subset-methods.html">`[`(<i>&lt;ChromatinContacts&gt;</i>,<i>&lt;GRanges&gt;</i>,<i>&lt;ANY&gt;</i>,<i>&lt;ANY&gt;</i>)</a></code> <code><a href="subset-methods.html">`[`(<i>&lt;ChromatinContacts&gt;</i>,<i>&lt;GInteractions&gt;</i>,<i>&lt;ANY&gt;</i>,<i>&lt;ANY&gt;</i>)</a></code> <code><a href="subset-methods.html">`[`(<i>&lt;ChromatinContacts&gt;</i>,<i>&lt;character&gt;</i>,<i>&lt;ANY&gt;</i>,<i>&lt;ANY&gt;</i>)</a></code>

        </dt>
        <dd>Subset ChromatinContacts objects by various criteria</dd>

        <dt>

          <code><a href="theme_hic.html">theme_hic()</a></code>

        </dt>
        <dd>ggplot2 theme optimized for Hi-C contact maps</dd>

        <dt>

          <code><a href="tidy.html">tidy()</a></code>

        </dt>
        <dd>Tidy hypergraph to long format</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Minghao Jiang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

