<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Resolution Depth Analysis for Hi-C/-like Data • gghic</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Resolution Depth Analysis for Hi-C/-like Data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gghic</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/gghic.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/depth.html">Resolution Depth Analysis for Hi-C/-like Data</a></li>
    <li><a class="dropdown-item" href="../articles/hypergraph.html">Multi-way Contact Analysis with Hypergraphs</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/jasonwong-lab/gghic/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Resolution Depth Analysis for Hi-C/-like Data</h1>
                        <h4 data-toc-skip class="author">Minghao
Jiang</h4>
            
            <h4 data-toc-skip class="date">2025-12-03</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jasonwong-lab/gghic/blob/master/vignettes/depth.Rmd" class="external-link"><code>vignettes/depth.Rmd</code></a></small>
      <div class="d-none name"><code>depth.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>This vignette demonstrates how to analyze Hi-C pairs data to find
optimal genomic resolutions using gghic’s resolution depth functions.
You’ll learn how to:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Calculate contact depth</strong> at different bin sizes</li>
<li>
<strong>Assess genome coverage</strong> across resolutions</li>
<li>
<strong>Find optimal resolution</strong> that balances detail and
completeness</li>
<li>
<strong>Visualize tradeoffs</strong> between resolution and
coverage</li>
</ol>
<div class="section level3">
<h3 id="when-to-use-these-functions">When to Use These Functions<a class="anchor" aria-label="anchor" href="#when-to-use-these-functions"></a>
</h3>
<p>These functions are essential for:</p>
<ul>
<li>
<strong>Pore-C &amp; long-read data</strong>: Determining the best
resolution for long DNA-DNA interactions</li>
<li>
<strong>Hi-C QC</strong>: Assessing data quality and sequencing
depth</li>
<li>
<strong>Resource planning</strong>: Finding the sweet spot before
expensive downstream analysis</li>
<li>
<strong>Publication standards</strong>: Communicating data
completeness to reviewers</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="getting-started">Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<p>Load required libraries:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">load_pkg</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">pkgs</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">for</span> <span class="op">(</span><span class="va">pkg</span> <span class="kw">in</span> <span class="va">pkgs</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">pkg</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">load_pkg</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"readr"</span>, <span class="st">"dplyr"</span>, <span class="st">"rappdirs"</span>, <span class="st">"gghic"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in library(package, lib.loc = lib.loc, character.only = TRUE,</span></span>
<span><span class="co">#&gt; logical.return = TRUE, : there is no package called 'readr'</span></span></code></pre></div>
<div class="section level3">
<h3 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h3>
<p>We’ll create synthetic pairs data for demonstration. In practice,
load from your contact matrix file:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create example pairs dataset</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></span>
<span><span class="va">n_interactions</span> <span class="op">&lt;-</span> <span class="fl">10000</span></span>
<span></span>
<span><span class="va">pairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  read_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"read_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">n_interactions</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  chrom1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr3"</span><span class="op">)</span>, <span class="va">n_interactions</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  pos1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1e6</span><span class="op">:</span><span class="fl">10e6</span>, <span class="va">n_interactions</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  chrom2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr3"</span><span class="op">)</span>, <span class="va">n_interactions</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  pos2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1e6</span><span class="op">:</span><span class="fl">10e6</span>, <span class="va">n_interactions</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">glimpse</span><span class="op">(</span><span class="va">pairs</span><span class="op">)</span></span>
<span><span class="co">#&gt; Rows: 10,000</span></span>
<span><span class="co">#&gt; Columns: 5</span></span>
<span><span class="co">#&gt; $ read_name <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "read_1"<span style="color: #949494;">, </span>"read_2"<span style="color: #949494;">, </span>"read_3"<span style="color: #949494;">, </span>"read_4"<span style="color: #949494;">, </span>"read_5"<span style="color: #949494;">, </span>"read_6"<span style="color: #949494;">, </span>…</span></span>
<span><span class="co">#&gt; $ chrom1    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "chr1"<span style="color: #949494;">, </span>"chr1"<span style="color: #949494;">, </span>"chr1"<span style="color: #949494;">, </span>"chr1"<span style="color: #949494;">, </span>"chr2"<span style="color: #949494;">, </span>"chr2"<span style="color: #949494;">, </span>"chr2"<span style="color: #949494;">, </span>"chr…</span></span>
<span><span class="co">#&gt; $ pos1      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 7501652<span style="color: #949494;">, </span>9834771<span style="color: #949494;">, </span>7511178<span style="color: #949494;">, </span>9988046<span style="color: #949494;">, </span>2376108<span style="color: #949494;">, </span>8056778<span style="color: #949494;">, </span>991689…</span></span>
<span><span class="co">#&gt; $ chrom2    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "chr3"<span style="color: #949494;">, </span>"chr1"<span style="color: #949494;">, </span>"chr1"<span style="color: #949494;">, </span>"chr2"<span style="color: #949494;">, </span>"chr1"<span style="color: #949494;">, </span>"chr2"<span style="color: #949494;">, </span>"chr3"<span style="color: #949494;">, </span>"chr…</span></span>
<span><span class="co">#&gt; $ pos2      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 4068533<span style="color: #949494;">, </span>1789229<span style="color: #949494;">, </span>6896113<span style="color: #949494;">, </span>6916211<span style="color: #949494;">, </span>7349759<span style="color: #949494;">, </span>9964631<span style="color: #949494;">, </span>150451…</span></span></code></pre></div>
<p><strong>Expected column format</strong>:</p>
<ul>
<li>
<code>read_name</code>: Unique identifier for each interaction</li>
<li>
<code>chrom1</code>, <code>chrom2</code>: Chromosome names (e.g.,
“chr1”)</li>
<li>
<code>pos1</code>, <code>pos2</code>: Genomic positions in base
pairs</li>
<li>Optional: strand, mapping quality, fragment ID (ignored by analysis
functions)</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="choosing-your-approach">Choosing Your Approach<a class="anchor" aria-label="anchor" href="#choosing-your-approach"></a>
</h2>
<p>The gghic package provides two complementary approaches for analyzing
pairs data:</p>
<table class="table">
<colgroup>
<col width="26%">
<col width="21%">
<col width="18%">
<col width="34%">
</colgroup>
<thead><tr class="header">
<th>Approach</th>
<th>Memory</th>
<th>Speed</th>
<th>When to Use</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<strong>In-memory</strong>
<code><a href="../reference/resolution-depth.html">calculateResolutionDepth()</a></code>
</td>
<td>O(interactions)</td>
<td>Fastest</td>
<td>Files &lt; 50% RAM, interactive analysis</td>
</tr>
<tr class="even">
<td>
<strong>Chunked</strong> <code><a href="../reference/resolution-depth.html">calcResDepthChunked()</a></code>
</td>
<td>O(unique bins)</td>
<td>Fast</td>
<td>Files &gt; 50% RAM, automated pipelines</td>
</tr>
</tbody>
</table>
<p><strong>In-Memory Processing (Smaller Files)</strong> - Load entire
dataset into memory - Single file read (fastest) - Typical memory: ~4
bytes per interaction - Best for: Interactive exploration,
visualization</p>
<p><strong>Chunked Streaming (Large Files)</strong> - Read file in
streaming fashion - Minimal memory footprint (~100 MB even for 100 GB
files) - C-accelerated file I/O - Automatic gzip decompression - Best
for: Production pipelines, server environments, very large datasets</p>
<p><strong>Decision Rule</strong>: If your file size &gt; available RAM
/ 2, use chunked functions.</p>
</div>
<div class="section level2">
<h2 id="step-1-calculate-resolution-depth">Step 1: Calculate Resolution Depth<a class="anchor" aria-label="anchor" href="#step-1-calculate-resolution-depth"></a>
</h2>
<p>The <code><a href="../reference/resolution-depth.html">calculateResolutionDepth()</a></code> function bins genomic
positions and counts unique interactions per bin:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate depth at 10kb resolution</span></span>
<span><span class="va">depth_10kb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">calculateResolutionDepth</a></span><span class="op">(</span><span class="va">pairs</span>, bin_size <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span></span>
<span></span>
<span><span class="va">depth_10kb</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2,699 × 3</span></span></span>
<span><span class="co">#&gt;    chrom   bin count</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1    845    12</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr2    812     6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr3    779     7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    846     7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr2    813     6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr3    780     4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1    847     7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr3    781     8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr2    814     6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr1    848     9</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2,689 more rows</span></span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Summary statistics</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">depth_10kb</span><span class="op">$</span><span class="va">count</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">#&gt;    1.00    5.00    7.00    7.41    9.00   19.00</span></span></code></pre></div>
<p><strong>What this returns</strong>:</p>
<ul>
<li>
<code>chrom</code>: Chromosome name</li>
<li>
<code>bin</code>: Bin number (position / bin_size, rounded up)</li>
<li>
<code>count</code>: Number of unique interactions in this bin</li>
</ul>
<p><strong>Interpreting the results</strong>:</p>
<ul>
<li>High counts = well-sequenced regions</li>
<li>Low counts = sparse or unmappable regions</li>
<li>Zero-count bins = gaps (excluded from output)</li>
</ul>
</div>
<div class="section level2">
<h2 id="step-2-assess-genome-coverage">Step 2: Assess Genome Coverage<a class="anchor" aria-label="anchor" href="#step-2-assess-genome-coverage"></a>
</h2>
<p>The <code><a href="../reference/resolution-depth.html">calculateGenomeCoverage()</a></code> function determines what
fraction of bins meet a minimum contact threshold:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate coverage at multiple resolutions</span></span>
<span><span class="va">resolutions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5e3</span>, <span class="fl">10e3</span>, <span class="fl">25e3</span>, <span class="fl">50e3</span>, <span class="fl">100e3</span>, <span class="fl">500e3</span><span class="op">)</span></span>
<span></span>
<span><span class="va">coverage_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  resolution_kb <span class="op">=</span> <span class="va">resolutions</span> <span class="op">/</span> <span class="fl">1e3</span>,</span>
<span>  n_bins <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  coverage <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">resolutions</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">calculateResolutionDepth</a></span><span class="op">(</span><span class="va">pairs</span>, <span class="va">resolutions</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="va">coverage_df</span><span class="op">$</span><span class="va">n_bins</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">depth</span><span class="op">)</span></span>
<span>  <span class="va">coverage_df</span><span class="op">$</span><span class="va">coverage</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">calculateGenomeCoverage</a></span><span class="op">(</span></span>
<span>    <span class="va">pairs</span>, <span class="va">resolutions</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>,</span>
<span>    min_contacts <span class="op">=</span> <span class="fl">50</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">coverage_df</span><span class="op">$</span><span class="va">coverage_pct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"%.1f%%"</span>, <span class="va">coverage_df</span><span class="op">$</span><span class="va">coverage</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="va">coverage_df</span></span>
<span><span class="co">#&gt;   resolution_kb n_bins   coverage coverage_pct</span></span>
<span><span class="co">#&gt; 1             5   5273 0.00000000         0.0%</span></span>
<span><span class="co">#&gt; 2            10   2699 0.00000000         0.0%</span></span>
<span><span class="co">#&gt; 3            25   1080 0.00000000         0.0%</span></span>
<span><span class="co">#&gt; 4            50    540 0.02777778         2.8%</span></span>
<span><span class="co">#&gt; 5           100    270 0.99629630        99.6%</span></span>
<span><span class="co">#&gt; 6           500     54 1.00000000       100.0%</span></span></code></pre></div>
<p><strong>Interpreting coverage</strong>:</p>
<ul>
<li>
<strong>High coverage (&gt;80%)</strong> at small bins:
Well-sequenced, good resolution potential</li>
<li>
<strong>Low coverage (&lt;50%)</strong> at large bins: Sparse data,
may need deeper sequencing</li>
<li>
<strong>Coverage curve shape</strong>: Shows sequencing depth and
data quality</li>
</ul>
</div>
<div class="section level2">
<h2 id="step-3-find-optimal-resolution">Step 3: Find Optimal Resolution<a class="anchor" aria-label="anchor" href="#step-3-find-optimal-resolution"></a>
</h2>
<p>Use binary search to find the <strong>smallest bin size</strong>
achieving your target coverage:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find resolution with ~80% coverage</span></span>
<span><span class="va">optimal_bin</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">findOptimalResolution</a></span><span class="op">(</span></span>
<span>  <span class="va">pairs</span>,</span>
<span>  min_bin <span class="op">=</span> <span class="fl">1e3</span>, <span class="co"># Start at 1 Kb</span></span>
<span>  max_bin <span class="op">=</span> <span class="fl">1e6</span>, <span class="co"># Search up to 1 Mb</span></span>
<span>  target_coverage <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  min_contacts <span class="op">=</span> <span class="fl">50</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Iteration 1: Testing bin size 500,500 bp: 98.18% coverage</span></span>
<span><span class="co">#&gt; Iteration 2: Testing bin size 250,749 bp: 97.30% coverage</span></span>
<span><span class="co">#&gt; Iteration 3: Testing bin size 125,874 bp: 97.72% coverage</span></span>
<span><span class="co">#&gt; Iteration 4: Testing bin size 63,436 bp: 36.60% coverage</span></span>
<span><span class="co">#&gt; Iteration 5: Testing bin size 94,655 bp: 97.92% coverage</span></span>
<span><span class="co">#&gt; Iteration 6: Testing bin size 79,045 bp: 88.41% coverage</span></span>
<span><span class="co">#&gt; Iteration 7: Testing bin size 71,240 bp: 64.83% coverage</span></span>
<span><span class="co">#&gt; Iteration 8: Testing bin size 75,142 bp: 77.13% coverage</span></span>
<span><span class="co">#&gt; Iteration 9: Testing bin size 77,093 bp: 83.85% coverage</span></span>
<span><span class="co">#&gt; Iteration 10: Testing bin size 76,117 bp: 81.23% coverage</span></span>
<span><span class="co">#&gt; Iteration 11: Testing bin size 75,629 bp: 78.61% coverage</span></span>
<span><span class="co">#&gt; Iteration 12: Testing bin size 75,873 bp: 81.23% coverage</span></span>
<span><span class="co">#&gt; Iteration 13: Testing bin size 75,751 bp: 79.55% coverage</span></span>
<span><span class="co">#&gt; Iteration 14: Testing bin size 75,812 bp: 79.27% coverage</span></span>
<span><span class="co">#&gt; Iteration 15: Testing bin size 75,842 bp: 80.67% coverage</span></span>
<span><span class="co">#&gt; Iteration 16: Testing bin size 75,827 bp: 80.39% coverage</span></span>
<span><span class="co">#&gt; Iteration 17: Testing bin size 75,819 bp: 79.83% coverage</span></span>
<span><span class="co">#&gt; Iteration 18: Testing bin size 75,823 bp: 80.11% coverage</span></span>
<span><span class="co">#&gt; Iteration 19: Testing bin size 75,821 bp: 79.55% coverage</span></span>
<span><span class="co">#&gt; Iteration 20: Testing bin size 75,822 bp: 79.55% coverage</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span></span>
<span>  <span class="st">"Optimal bin size: %d bp (%.1f Kb)\n"</span>, <span class="va">optimal_bin</span>, <span class="va">optimal_bin</span> <span class="op">/</span> <span class="fl">1000</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Optimal bin size: 75823 bp (75.8 Kb)</span></span>
<span></span>
<span><span class="co"># Verify the result</span></span>
<span><span class="va">actual_coverage</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">calculateGenomeCoverage</a></span><span class="op">(</span></span>
<span>  <span class="va">pairs</span>, <span class="va">optimal_bin</span>,</span>
<span>  min_contacts <span class="op">=</span> <span class="fl">50</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"Actual coverage: %.2f%%\n"</span>, <span class="va">actual_coverage</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Actual coverage: 80.11%</span></span></code></pre></div>
<p><strong>How it works</strong>:</p>
<ol style="list-style-type: decimal">
<li>Binary search across the bin size range</li>
<li>Tests each candidate bin size</li>
<li>Returns smallest bin that meets target coverage</li>
<li>Typically completes in 10-15 iterations</li>
</ol>
<p><strong>Pro tip</strong>: Start with
<code>target_coverage = 0.8</code> (80%). Adjust based on your analysis
needs.</p>
</div>
<div class="section level2">
<h2 id="step-4-visualize-coverage-distribution">Step 4: Visualize Coverage Distribution<a class="anchor" aria-label="anchor" href="#step-4-visualize-coverage-distribution"></a>
</h2>
<p>Create a boxplot showing how contact counts vary across
resolutions:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotResolutionCoverage.html">plotResolutionCoverage</a></span><span class="op">(</span></span>
<span>  <span class="va">pairs</span>,</span>
<span>  bin_sizes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5e3</span>, <span class="fl">10e3</span>, <span class="fl">25e3</span>, <span class="fl">50e3</span>, <span class="fl">100e3</span>, <span class="fl">500e3</span><span class="op">)</span>,</span>
<span>  min_contacts <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"Contact Distribution by Resolution"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="depth_files/figure-html/unnamed-chunk-7-1.png" class="r-plt" alt="Resolution depth analysis plot" width="100%"><p class="caption">
Distribution of contacts per bin at different resolutions
</p>
</div>
<p><strong>What to look for</strong>:</p>
<ul>
<li>
<strong>Widening boxes</strong>: Decreasing contacts with larger
bins (expected)</li>
<li>
<strong>Outliers (dots)</strong>: Unusually enriched regions (e.g.,
centromeres, pericentromeres)</li>
<li>
<strong>Median position</strong>: Typical coverage at that
resolution</li>
<li>
<strong>Whisker length</strong>: Range of contact values (high
whiskers = variable coverage)</li>
</ul>
</div>
<div class="section level2">
<h2 id="step-5-plot-coverage-vs-resolution">Step 5: Plot Coverage vs Resolution<a class="anchor" aria-label="anchor" href="#step-5-plot-coverage-vs-resolution"></a>
</h2>
<p>Create a line plot showing how coverage changes with resolution:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCoverageCurve.html">plotCoverageCurve</a></span><span class="op">(</span></span>
<span>  <span class="va">pairs</span>,</span>
<span>  bin_sizes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5e3</span>, <span class="fl">10e3</span>, <span class="fl">25e3</span>, <span class="fl">50e3</span>, <span class="fl">100e3</span>, <span class="fl">500e3</span>, <span class="fl">1e6</span><span class="op">)</span>,</span>
<span>  min_contacts <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"Coverage vs Resolution"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="depth_files/figure-html/unnamed-chunk-8-1.png" class="r-plt" alt="Resolution depth analysis plot" width="100%"><p class="caption">
Genome coverage as a function of bin size
</p>
</div>
<p><strong>Curve interpretation</strong>:</p>
<ul>
<li>
<strong>Sharp drop at small bins</strong>: Limited sequencing
depth</li>
<li>
<strong>Plateau region</strong>: Where additional bin size doesn’t
improve coverage much</li>
<li>
<strong>“Knee” of curve</strong>: The optimal resolution sweet
spot</li>
<li>
<strong>Steep regions</strong>: Good resolution potential at that
bin size</li>
</ul>
</div>
<div class="section level2">
<h2 id="processing-large-files-100-gb">Processing Large Files (100 GB+)<a class="anchor" aria-label="anchor" href="#processing-large-files-100-gb"></a>
</h2>
<blockquote>
<p><strong>⚠️ Note</strong>: The code examples in this section are
provided for reference but are <strong>not executed</strong> in this
vignette.</p>
</blockquote>
<p>For files that don’t fit in R memory, use the <code>_chunked</code>
variants which stream the file with C-level I/O:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Download example file with caching</span></span>
<span><span class="va">cache_dir</span> <span class="op">&lt;-</span> <span class="fu">rappdirs</span><span class="fu">::</span><span class="fu"><a href="https://rappdirs.r-lib.org/reference/user_cache_dir.html" class="external-link">user_cache_dir</a></span><span class="op">(</span><span class="st">"gghic"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">cache_dir</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span>, showWarnings <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pairs_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">cache_dir</span>, <span class="st">"test.txt.gz"</span><span class="op">)</span></span>
<span><span class="va">download_url</span> <span class="op">&lt;-</span> <span class="st">"https://www.dropbox.com/scl/fi/yc4axg1mf2i9zylg3d0oe/test.txt.gz?rlkey=sdsdhsnixo01koo38d242y4c8&amp;st=t4rn0js0&amp;dl=1"</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">pairs_file</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"Downloading test data to cache directory..."</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span></span>
<span>    <span class="va">download_url</span>, <span class="va">pairs_file</span>, method <span class="op">=</span> <span class="st">"auto"</span>, quiet <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"Downloaded to: "</span>, <span class="va">pairs_file</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"Using cached file: "</span>, <span class="va">pairs_file</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="section level3">
<h3 id="approach-1-direct-file-reading-simple-but-slower">Approach 1: Direct File Reading (Simple but Slower)<a class="anchor" aria-label="anchor" href="#approach-1-direct-file-reading-simple-but-slower"></a>
</h3>
<p>Read the file each time a function is called:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Each function reads the file independently</span></span>
<span><span class="va">opt_bin</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">findOptResChunked</a></span><span class="op">(</span></span>
<span>  pairs_file <span class="op">=</span> <span class="va">pairs_file</span>,</span>
<span>  min_bin <span class="op">=</span> <span class="fl">1e3</span>,</span>
<span>  max_bin <span class="op">=</span> <span class="fl">5e4</span>,</span>
<span>  target_coverage <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  min_contacts <span class="op">=</span> <span class="fl">1000</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Reads file again</span></span>
<span><span class="va">coverage</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">calcGenomeCovChunked</a></span><span class="op">(</span></span>
<span>  pairs_file <span class="op">=</span> <span class="va">pairs_file</span>,</span>
<span>  bin_size <span class="op">=</span> <span class="va">opt_bin</span>,</span>
<span>  min_contacts <span class="op">=</span> <span class="fl">1000</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Reads file again</span></span>
<span><span class="va">depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">calcResDepthChunked</a></span><span class="op">(</span></span>
<span>  pairs_file <span class="op">=</span> <span class="va">pairs_file</span>,</span>
<span>  bin_size <span class="op">=</span> <span class="va">opt_bin</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><strong>Problem</strong>: For a 10 GB file, you’re reading 10 GB
three times (30 GB total I/O)!</p>
</div>
<div class="section level3">
<h3 id="approach-2-cache-once-reuse-many-times-fastest">Approach 2: Cache Once, Reuse Many Times (FASTEST!)<a class="anchor" aria-label="anchor" href="#approach-2-cache-once-reuse-many-times-fastest"></a>
</h3>
<p>Read the file once into C memory, then reuse the cached data:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. READ FILE INTO C MEMORY ONCE ----</span></span>
<span><span class="co"># This takes time initially but creates a persistent cache</span></span>
<span><span class="va">cache</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">readPairsCache</a></span><span class="op">(</span><span class="va">pairs_file</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"File loaded into C memory cache\n"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. FIND OPTIMAL RESOLUTION (using cache) ----</span></span>
<span><span class="co"># This is now instant - no file I/O!</span></span>
<span><span class="va">opt_bin</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">findOptResChunked</a></span><span class="op">(</span></span>
<span>  cache <span class="op">=</span> <span class="va">cache</span>,</span>
<span>  min_bin <span class="op">=</span> <span class="fl">1e3</span>,</span>
<span>  max_bin <span class="op">=</span> <span class="fl">5e4</span>,</span>
<span>  target_coverage <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  min_contacts <span class="op">=</span> <span class="fl">1000</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Optimal resolution:"</span>, <span class="va">opt_bin</span> <span class="op">/</span> <span class="fl">1e3</span>, <span class="st">"Kb\n"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 3. CHECK COVERAGE (using cache) ----</span></span>
<span><span class="co"># Also instant!</span></span>
<span><span class="va">coverage</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">calcGenomeCovChunked</a></span><span class="op">(</span></span>
<span>  cache <span class="op">=</span> <span class="va">cache</span>,</span>
<span>  bin_size <span class="op">=</span> <span class="va">opt_bin</span>,</span>
<span>  min_contacts <span class="op">=</span> <span class="fl">1000</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Coverage:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"%.1f%%\n"</span>, <span class="va">coverage</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 4. EXTRACT BINNED DEPTHS (using cache) ----</span></span>
<span><span class="co"># Still instant!</span></span>
<span><span class="va">depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">calcResDepthChunked</a></span><span class="op">(</span></span>
<span>  cache <span class="op">=</span> <span class="va">cache</span>,</span>
<span>  bin_size <span class="op">=</span> <span class="va">opt_bin</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 5. TRY DIFFERENT BIN SIZES (no additional I/O!) ----</span></span>
<span><span class="va">depth_5kb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">calcResDepthChunked</a></span><span class="op">(</span>cache <span class="op">=</span> <span class="va">cache</span>, bin_size <span class="op">=</span> <span class="fl">5000</span><span class="op">)</span></span>
<span><span class="va">depth_10kb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">calcResDepthChunked</a></span><span class="op">(</span>cache <span class="op">=</span> <span class="va">cache</span>, bin_size <span class="op">=</span> <span class="fl">10000</span><span class="op">)</span></span>
<span><span class="va">depth_50kb</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">calcResDepthChunked</a></span><span class="op">(</span>cache <span class="op">=</span> <span class="va">cache</span>, bin_size <span class="op">=</span> <span class="fl">50000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Cache is automatically freed when removed or R session ends</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">cache</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/gc.html" class="external-link">gc</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><strong>Performance</strong>: For a 10 GB file: - Without cache: ~30
seconds × 3 operations = <strong>90 seconds</strong> - With cache: ~30
seconds (once) + ~0.1 seconds × operations = <strong>30
seconds</strong></p>
<p><strong>3x faster!</strong> And the speedup increases with more
operations.</p>
</div>
<div class="section level3">
<h3 id="approach-3-get-cache-from-findoptreschunked">Approach 3: Get Cache from findOptResChunked<a class="anchor" aria-label="anchor" href="#approach-3-get-cache-from-findoptreschunked"></a>
</h3>
<p>Let <code><a href="../reference/resolution-depth.html">findOptResChunked()</a></code> return the cache for subsequent
use:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find optimal resolution and get cache back</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">findOptResChunked</a></span><span class="op">(</span></span>
<span>  pairs_file <span class="op">=</span> <span class="va">pairs_file</span>,</span>
<span>  min_bin <span class="op">=</span> <span class="fl">1e3</span>,</span>
<span>  max_bin <span class="op">=</span> <span class="fl">5e4</span>,</span>
<span>  target_coverage <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  min_contacts <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  return_cache <span class="op">=</span> <span class="cn">TRUE</span> <span class="co"># Get the cache back!</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Extract results</span></span>
<span><span class="va">opt_bin</span> <span class="op">&lt;-</span> <span class="va">result</span><span class="op">$</span><span class="va">bin_size</span></span>
<span><span class="va">cache</span> <span class="op">&lt;-</span> <span class="va">result</span><span class="op">$</span><span class="va">cache</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Optimal resolution:"</span>, <span class="va">opt_bin</span> <span class="op">/</span> <span class="fl">1e3</span>, <span class="st">"Kb\n"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Now reuse the cache for other operations</span></span>
<span><span class="va">coverage</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">calcGenomeCovChunked</a></span><span class="op">(</span></span>
<span>  cache <span class="op">=</span> <span class="va">cache</span>,</span>
<span>  bin_size <span class="op">=</span> <span class="va">opt_bin</span>,</span>
<span>  min_contacts <span class="op">=</span> <span class="fl">1000</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/resolution-depth.html">calcResDepthChunked</a></span><span class="op">(</span></span>
<span>  cache <span class="op">=</span> <span class="va">cache</span>,</span>
<span>  bin_size <span class="op">=</span> <span class="va">opt_bin</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="when-to-use-each-approach">When to Use Each Approach<a class="anchor" aria-label="anchor" href="#when-to-use-each-approach"></a>
</h3>
<table class="table">
<colgroup>
<col width="27%">
<col width="27%">
<col width="44%">
</colgroup>
<thead><tr class="header">
<th>Approach</th>
<th>Best For</th>
<th>I/O Operations</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Direct File</strong></td>
<td>Single analysis, simple scripts</td>
<td>N operations</td>
</tr>
<tr class="even">
<td><strong>Explicit Cache</strong></td>
<td>Interactive exploration, multiple analyses</td>
<td>1 operation</td>
</tr>
<tr class="odd">
<td><strong>Return Cache</strong></td>
<td>When starting with <code><a href="../reference/resolution-depth.html">findOptResChunked()</a></code>
</td>
<td>1 operation</td>
</tr>
</tbody>
</table>
<p><strong>Rule of thumb</strong>: If you’ll call more than one chunked
function, use the cache!</p>
</div>
</div>
<div class="section level2">
<h2 id="interpreting-your-results">Interpreting Your Results<a class="anchor" aria-label="anchor" href="#interpreting-your-results"></a>
</h2>
<div class="section level3">
<h3 id="what-the-coverage-curve-tells-you">What the Coverage Curve Tells You<a class="anchor" aria-label="anchor" href="#what-the-coverage-curve-tells-you"></a>
</h3>
<p><strong>Sharp drop at small bins</strong> - Data is sparse at high
resolution - Limited sequencing depth or short reads - Consider deeper
sequencing or larger bins</p>
<p><strong>Plateau region</strong> - Coverage maxes out (can’t improve
by increasing bin size) - Indicates natural sequencing depth ceiling -
Optimal resolution likely at transition point</p>
<p><strong>High coverage at large bins (&gt;80%)</strong> -
Well-sequenced dataset - Can use smaller bins safely - Good candidate
for publication-quality analysis</p>
<p><strong>Low coverage at large bins (&lt;50%)</strong> - Sparse data -
May need additional sequencing - Consider quality filtering</p>
</div>
<div class="section level3">
<h3 id="choosing-target-coverage">Choosing Target Coverage<a class="anchor" aria-label="anchor" href="#choosing-target-coverage"></a>
</h3>
<ul>
<li>
<strong>Exploration</strong>: 50-60% (quick, uses fewer bins)</li>
<li>
<strong>Standard analysis</strong>: 75-85% (good balance)</li>
<li>
<strong>Publication</strong>: 85-90%+ (most stringent)</li>
</ul>
<p>Higher coverage = higher confidence but requires more sequencing.</p>
</div>
</div>
<div class="section level2">
<h2 id="next-steps">Next Steps<a class="anchor" aria-label="anchor" href="#next-steps"></a>
</h2>
<p>Once you’ve found your optimal resolution, explore other gghic
functions for visualization:</p>
<ul>
<li>
<code><a href="../reference/geom_hic.html">geom_hic()</a></code>: Create interaction scatter plots</li>
<li>
<code><a href="../reference/geom_loop.html">geom_loop()</a></code>: Highlight chromatin loops</li>
<li>
<code><a href="../reference/geom_tad.html">geom_tad()</a></code>: Show topologically associating domains</li>
<li>
<code><a href="../reference/geom_ideogram.html">geom_ideogram()</a></code>: Add chromosome context</li>
<li>
<code><a href="../reference/theme_hic.html">theme_hic()</a></code>: Specialized theme for Hi-C plots</li>
</ul>
<p>See the main gghic vignette for complete examples.</p>
</div>
<div class="section level2">
<h2 id="faq">FAQ<a class="anchor" aria-label="anchor" href="#faq"></a>
</h2>
<p><strong>Q: Why does coverage decrease with larger bins?</strong> A:
By definition—larger bins aggregate more positions but contact counts
don’t increase proportionally. Coverage measures the fraction of bins
with sufficient contacts.</p>
<p><strong>Q: Should I use chunked functions for small files?</strong>
A: No, use in-memory functions—they’re simpler and faster for files that
fit in RAM.</p>
<p><strong>Q: Can I use these functions on non-Hi-C data?</strong> A:
Yes, any “pairs” data with (chrom, position) coordinates works: Pore-C,
4C, 5C, etc.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.2 (2025-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] gghic_0.2.0    rappdirs_0.3.3 dplyr_1.1.4   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3          rstudioapi_0.17.1          </span></span>
<span><span class="co">#&gt;   [3] jsonlite_2.0.0              magrittr_2.0.4             </span></span>
<span><span class="co">#&gt;   [5] GenomicFeatures_1.62.0      farver_2.1.2               </span></span>
<span><span class="co">#&gt;   [7] rmarkdown_2.30              fs_1.6.6                   </span></span>
<span><span class="co">#&gt;   [9] BiocIO_1.20.0               ragg_1.5.0                 </span></span>
<span><span class="co">#&gt;  [11] vctrs_0.6.5                 memoise_2.0.1              </span></span>
<span><span class="co">#&gt;  [13] Rsamtools_2.26.0            RCurl_1.98-1.17            </span></span>
<span><span class="co">#&gt;  [15] base64enc_0.1-3             htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt;  [17] S4Arrays_1.10.0             progress_1.2.3             </span></span>
<span><span class="co">#&gt;  [19] curl_7.0.0                  Rhdf5lib_1.32.0            </span></span>
<span><span class="co">#&gt;  [21] SparseArray_1.10.3          Formula_1.2-5              </span></span>
<span><span class="co">#&gt;  [23] rhdf5_2.54.0                sass_0.4.10                </span></span>
<span><span class="co">#&gt;  [25] bslib_0.9.0                 htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt;  [27] desc_1.4.3                  Gviz_1.54.0                </span></span>
<span><span class="co">#&gt;  [29] httr2_1.2.1                 cachem_1.1.0               </span></span>
<span><span class="co">#&gt;  [31] GenomicAlignments_1.46.0    lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt;  [33] pkgconfig_2.0.3             Matrix_1.7-4               </span></span>
<span><span class="co">#&gt;  [35] R6_2.6.1                    fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [37] MatrixGenerics_1.22.0       digest_0.6.39              </span></span>
<span><span class="co">#&gt;  [39] colorspace_2.1-2            AnnotationDbi_1.72.0       </span></span>
<span><span class="co">#&gt;  [41] S4Vectors_0.48.0            textshaping_1.0.4          </span></span>
<span><span class="co">#&gt;  [43] Hmisc_5.2-4                 GenomicRanges_1.62.0       </span></span>
<span><span class="co">#&gt;  [45] RSQLite_2.4.5               labeling_0.4.3             </span></span>
<span><span class="co">#&gt;  [47] filelock_1.0.3              httr_1.4.7                 </span></span>
<span><span class="co">#&gt;  [49] abind_1.4-8                 compiler_4.5.2             </span></span>
<span><span class="co">#&gt;  [51] withr_3.0.2                 bit64_4.6.0-1              </span></span>
<span><span class="co">#&gt;  [53] htmlTable_2.4.3             S7_0.2.1                   </span></span>
<span><span class="co">#&gt;  [55] backports_1.5.0             BiocParallel_1.44.0        </span></span>
<span><span class="co">#&gt;  [57] DBI_1.2.3                   biomaRt_2.66.0             </span></span>
<span><span class="co">#&gt;  [59] DelayedArray_0.36.0         rjson_0.2.23               </span></span>
<span><span class="co">#&gt;  [61] tools_4.5.2                 foreign_0.8-90             </span></span>
<span><span class="co">#&gt;  [63] nnet_7.3-20                 glue_1.8.0                 </span></span>
<span><span class="co">#&gt;  [65] restfulr_0.0.16             InteractionSet_1.38.0      </span></span>
<span><span class="co">#&gt;  [67] rhdf5filters_1.22.0         grid_4.5.2                 </span></span>
<span><span class="co">#&gt;  [69] checkmate_2.3.3             cluster_2.1.8.1            </span></span>
<span><span class="co">#&gt;  [71] generics_0.1.4              gtable_0.3.6               </span></span>
<span><span class="co">#&gt;  [73] BSgenome_1.78.0             tidyr_1.3.1                </span></span>
<span><span class="co">#&gt;  [75] ensembldb_2.34.0            data.table_1.17.8          </span></span>
<span><span class="co">#&gt;  [77] hms_1.1.4                   utf8_1.2.6                 </span></span>
<span><span class="co">#&gt;  [79] XVector_0.50.0              BiocGenerics_0.56.0        </span></span>
<span><span class="co">#&gt;  [81] pillar_1.11.1               stringr_1.6.0              </span></span>
<span><span class="co">#&gt;  [83] BiocFileCache_3.0.0         lattice_0.22-7             </span></span>
<span><span class="co">#&gt;  [85] rtracklayer_1.70.0          bit_4.6.0                  </span></span>
<span><span class="co">#&gt;  [87] deldir_2.0-4                biovizBase_1.58.0          </span></span>
<span><span class="co">#&gt;  [89] tidyselect_1.2.1            Biostrings_2.78.0          </span></span>
<span><span class="co">#&gt;  [91] knitr_1.50                  gridExtra_2.3              </span></span>
<span><span class="co">#&gt;  [93] IRanges_2.44.0              Seqinfo_1.0.0              </span></span>
<span><span class="co">#&gt;  [95] ProtGenerics_1.42.0         SummarizedExperiment_1.40.0</span></span>
<span><span class="co">#&gt;  [97] stats4_4.5.2                xfun_0.54                  </span></span>
<span><span class="co">#&gt;  [99] Biobase_2.70.0              matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt; [101] stringi_1.8.7               UCSC.utils_1.6.0           </span></span>
<span><span class="co">#&gt; [103] lazyeval_0.2.2              yaml_2.3.11                </span></span>
<span><span class="co">#&gt; [105] evaluate_1.0.5              codetools_0.2-20           </span></span>
<span><span class="co">#&gt; [107] cigarillo_1.0.0             interp_1.1-6               </span></span>
<span><span class="co">#&gt; [109] tibble_3.3.0                cli_3.6.5                  </span></span>
<span><span class="co">#&gt; [111] rpart_4.1.24                systemfonts_1.3.1          </span></span>
<span><span class="co">#&gt; [113] jquerylib_0.1.4             dichromat_2.0-0.1          </span></span>
<span><span class="co">#&gt; [115] Rcpp_1.1.0                  GenomeInfoDb_1.46.1        </span></span>
<span><span class="co">#&gt; [117] dbplyr_2.5.1                png_0.1-8                  </span></span>
<span><span class="co">#&gt; [119] XML_3.99-0.20               parallel_4.5.2             </span></span>
<span><span class="co">#&gt; [121] pkgdown_2.2.0               ggplot2_4.0.1              </span></span>
<span><span class="co">#&gt; [123] blob_1.2.4                  prettyunits_1.2.0          </span></span>
<span><span class="co">#&gt; [125] latticeExtra_0.6-31         jpeg_0.1-11                </span></span>
<span><span class="co">#&gt; [127] AnnotationFilter_1.34.0     bitops_1.0-9               </span></span>
<span><span class="co">#&gt; [129] txdbmaker_1.6.0             VariantAnnotation_1.56.0   </span></span>
<span><span class="co">#&gt; [131] scales_1.4.0                purrr_1.2.0                </span></span>
<span><span class="co">#&gt; [133] crayon_1.5.3                rlang_1.1.6                </span></span>
<span><span class="co">#&gt; [135] KEGGREST_1.50.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Minghao Jiang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
