<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Use gghic to Visualise Genomic Interaction Data • gghic</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Use gghic to Visualise Genomic Interaction Data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gghic</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/gghic.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/jasonwong-lab/gghic/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Use gghic to Visualise Genomic Interaction Data</h1>
            
            <h4 data-toc-skip class="date">2025-05-21</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jasonwong-lab/gghic/blob/master/vignettes/gghic.Rmd" class="external-link"><code>vignettes/gghic.Rmd</code></a></small>
      <div class="d-none name"><code>gghic.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><strong>gghic</strong> is a versatile R package for creating
flexible, publication-ready visualizations of 3D genome organization
data. With gghic, you can easily explore and present Hi-C contact maps,
topologically associating domains (TADs), chromatin loops, gene
annotations, and other genomic features in a unified and customizable
framework.</p>
<p>Whether you are analyzing large-scale Hi-C experiments, visualizing
multi-chromosome interactions, or integrating additional genomic tracks
(such as ChIP-seq or BigWig data), gghic provides a tidyverse-friendly
and extensible toolkit for your research.</p>
<p>This tutorial will guide you through the core features of gghic, from
data preparation to advanced visualization. By the end, you’ll be able
to generate high-quality, informative figures for your genomics projects
with just a few lines of code.</p>
<blockquote>
<p><strong>Learn more:</strong> - <a href="https://arxiv.org/abs/2412.03005" class="external-link">arXiv preprint</a></p>
</blockquote>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="section level3">
<h3 id="load-required-packages">Load Required Packages<a class="anchor" aria-label="anchor" href="#load-required-packages"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">load_pkg</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">pkgs</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">for</span> <span class="op">(</span><span class="va">pkg</span> <span class="kw">in</span> <span class="va">pkgs</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">pkg</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">load_pkg</span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"gghic"</span>, <span class="st">"ggplot2"</span>, <span class="st">"tibble"</span>, <span class="st">"scales"</span>, <span class="st">"dplyr"</span>, <span class="st">"tidyr"</span>, <span class="st">"glue"</span>,</span>
<span>    <span class="st">"HiCExperiment"</span>, <span class="st">"GenomicRanges"</span>, <span class="st">"InteractionSet"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="download-example-data">Download Example data<a class="anchor" aria-label="anchor" href="#download-example-data"></a>
</h3>
<p>Example data files are hosted at gghic-data to keep the package
lightweight.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">download_example_files</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">cache_dir</span>, <span class="va">check_exists</span> <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">cache_dir</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">cache_dir</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    <span class="st">"chr4_11-100kb.cool"</span> <span class="op">=</span> <span class="st">"cooler/chr4_11-100kb.cool"</span>,</span>
<span>    <span class="st">"chr4_11-5kb.cool"</span> <span class="op">=</span> <span class="st">"cooler/chr4_11-5kb.cool"</span>,</span>
<span>    <span class="st">"track1.bigWig"</span> <span class="op">=</span> <span class="st">"bigwig/track1.bigWig"</span>,</span>
<span>    <span class="st">"track2.bigWig"</span> <span class="op">=</span> <span class="st">"bigwig/track2.bigWig"</span>,</span>
<span>    <span class="st">"gencode-chr4_11.gtf.gz"</span> <span class="op">=</span> <span class="st">"gtf/gencode-chr4_11.gtf.gz"</span>,</span>
<span>    <span class="st">"TADs_500kb-chr4_11.tsv"</span> <span class="op">=</span> <span class="st">"tad/TADs_500kb-chr4_11.tsv"</span>,</span>
<span>    <span class="st">"loops-chr4_11.txt"</span> <span class="op">=</span> <span class="st">"loop/loops-chr4_11.txt"</span>,</span>
<span>    <span class="st">"gis_hic.rds"</span> <span class="op">=</span> <span class="st">"multiway/gis_hic.rds"</span>,</span>
<span>    <span class="st">"concatemers.rds"</span> <span class="op">=</span> <span class="st">"multiway/concatemers.rds"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span>  <span class="va">url_base</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>    <span class="st">"https://raw.githubusercontent.com/"</span>,</span>
<span>    <span class="st">"mhjiang97/gghic-data/refs/heads/master/"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span>  <span class="kw">for</span> <span class="op">(</span><span class="va">file_name</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">files</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">cache_dir</span>, <span class="va">file_name</span><span class="op">)</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">check_exists</span> <span class="op">&amp;&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">file_path</span><span class="op">)</span><span class="op">)</span> <span class="kw">next</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">url_base</span>, <span class="va">files</span><span class="op">[[</span><span class="va">file_name</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, <span class="va">file_path</span>, method <span class="op">=</span> <span class="st">"curl"</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dir_cache</span> <span class="op">&lt;-</span> <span class="st">"../data"</span></span>
<span></span>
<span><span class="fu">download_example_files</span><span class="op">(</span><span class="va">dir_cache</span><span class="op">)</span></span>
<span></span>
<span><span class="va">path_cf_100</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir_cache</span>, <span class="st">"chr4_11-100kb.cool"</span><span class="op">)</span></span>
<span><span class="va">path_cf_5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir_cache</span>, <span class="st">"chr4_11-5kb.cool"</span><span class="op">)</span></span>
<span><span class="va">path_gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir_cache</span>, <span class="st">"gencode-chr4_11.gtf.gz"</span><span class="op">)</span></span>
<span><span class="va">paths_track</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir_cache</span>, <span class="fu">glue</span><span class="fu">::</span><span class="fu"><a href="https://glue.tidyverse.org/reference/glue.html" class="external-link">glue</a></span><span class="op">(</span><span class="st">"track{1:2}.bigWig"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">path_tad</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir_cache</span>, <span class="st">"TADs_500kb-chr4_11.tsv"</span><span class="op">)</span></span>
<span><span class="va">path_loop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir_cache</span>, <span class="st">"loops-chr4_11.txt"</span><span class="op">)</span></span>
<span><span class="va">path_gis_hic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir_cache</span>, <span class="st">"gis_hic.rds"</span><span class="op">)</span></span>
<span><span class="va">path_concatemers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">dir_cache</span>, <span class="st">"concatemers.rds"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="data-preparation">Data Preparation<a class="anchor" aria-label="anchor" href="#data-preparation"></a>
</h2>
<div class="section level3">
<h3 id="import-hi-c-data">Import Hi-C Data<a class="anchor" aria-label="anchor" href="#import-hi-c-data"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic_100</span> <span class="op">&lt;-</span> <span class="va">path_cf_100</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">HiCExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiCExperiment/man/CoolFile-class.html" class="external-link">CoolFile</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">HiCExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocIO/man/IO.html" class="external-link">import</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hic_5</span> <span class="op">&lt;-</span> <span class="va">path_cf_5</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">HiCExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiCExperiment/man/CoolFile-class.html" class="external-link">CoolFile</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">HiCExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocIO/man/IO.html" class="external-link">import</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="prepare-data-for-visualisation">Prepare Data for Visualisation<a class="anchor" aria-label="anchor" href="#prepare-data-for-visualisation"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scale_data</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">data</span>, <span class="va">score_column</span> <span class="op">=</span> <span class="st">"balanced"</span>, <span class="va">scale_method</span> <span class="op">=</span> <span class="va">log10</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">inherits</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"HiCExperiment"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">gis</span> <span class="op">&lt;-</span> <span class="fu">InteractionSet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/interaction-accessors.html" class="external-link">interactions</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span>
<span>  <span class="op">}</span> <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">inherits</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"GInteractions"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">gis</span> <span class="op">&lt;-</span> <span class="va">data</span></span>
<span>  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"Input data must be a HiCExperiment or GInteractions object."</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span></span>
<span>  <span class="va">x</span> <span class="op">&lt;-</span> <span class="va">gis</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>score <span class="op">=</span> <span class="fu">scale_method</span><span class="op">(</span><span class="va">.data</span><span class="op">[[</span><span class="va">score_column</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">score</span><span class="op">)</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/is.finite.html" class="external-link">is.infinite</a></span><span class="op">(</span><span class="va">score</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>score <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/oob.html" class="external-link">oob_squish</a></span><span class="op">(</span><span class="va">score</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">score</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">score</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">x</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x_100</span> <span class="op">&lt;-</span> <span class="fu">scale_data</span><span class="op">(</span><span class="va">hic_100</span><span class="op">)</span></span>
<span><span class="va">x_5</span> <span class="op">&lt;-</span> <span class="fu">scale_data</span><span class="op">(</span><span class="va">hic_5</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualisations">Visualisations<a class="anchor" aria-label="anchor" href="#visualisations"></a>
</h2>
<div class="section level3">
<h3 id="hi-c-heatmap">Hi-C Heatmap<a class="anchor" aria-label="anchor" href="#hi-c-heatmap"></a>
</h3>
<p><code><a href="../reference/geom_hic.html">geom_hic()</a></code> is the main function to visualise Hi-C/-like
data. It requires the following aesthetics: <code>seqnames1</code>,
<code>start1</code>, <code>end1</code>, <code>seqnames2</code>,
<code>start2</code>, <code>end2</code>, and <code>fill</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">x_100</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames1</span> <span class="op">==</span> <span class="st">"chr11"</span>, <span class="va">seqnames2</span> <span class="op">==</span> <span class="st">"chr11"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/drop_na.html" class="external-link">drop_na</a></span><span class="op">(</span><span class="va">score</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>      seqnames1 <span class="op">=</span> <span class="va">seqnames1</span>, start1 <span class="op">=</span> <span class="va">start1</span>, end1 <span class="op">=</span> <span class="va">end1</span>,</span>
<span>      seqnames2 <span class="op">=</span> <span class="va">seqnames2</span>, start2 <span class="op">=</span> <span class="va">start2</span>, end2 <span class="op">=</span> <span class="va">end2</span>,</span>
<span>      fill <span class="op">=</span> <span class="va">score</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_hic.html">geom_hic</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p><code><a href="../reference/theme_hic.html">theme_hic()</a></code> is a theme function for
<code><a href="../reference/geom_hic.html">geom_hic()</a></code> that enhances the appearance of the plot.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="../reference/theme_hic.html">theme_hic</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p><code><a href="../reference/gghic.html">gghic()</a></code> is a wrapper function that can take either a
<code>HiCExperiment</code> object, a <code>GenomicInteractions</code>
object, or a <code>tibble</code>/<code>data.frame</code> as input.
(exactly the same as above).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic_100</span><span class="op">[</span><span class="st">"chr11"</span><span class="op">]</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/gghic.html">gghic</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="ideogram">Ideogram<a class="anchor" aria-label="anchor" href="#ideogram"></a>
</h3>
<p><code><a href="../reference/geom_ideogram.html">geom_ideogram()</a></code> can be used to add chromosome ideograms
to the plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x_100</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames1</span> <span class="op">==</span> <span class="st">"chr11"</span>, <span class="va">seqnames2</span> <span class="op">==</span> <span class="st">"chr11"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>      seqnames1 <span class="op">=</span> <span class="va">seqnames1</span>, start1 <span class="op">=</span> <span class="va">start1</span>, end1 <span class="op">=</span> <span class="va">end1</span>,</span>
<span>      seqnames2 <span class="op">=</span> <span class="va">seqnames2</span>, start2 <span class="op">=</span> <span class="va">start2</span>, end2 <span class="op">=</span> <span class="va">end2</span>,</span>
<span>      fill <span class="op">=</span> <span class="va">score</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_hic.html">geom_hic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/theme_hic.html">theme_hic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_ideogram.html">geom_ideogram</a></span><span class="op">(</span></span>
<span>    genome <span class="op">=</span> <span class="st">"hg19"</span>, highlight <span class="op">=</span> <span class="cn">FALSE</span>, length_ratio <span class="op">=</span> <span class="fl">0.7</span>, fontsize <span class="op">=</span> <span class="fl">8</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; Warning in getMethods(coerce, table = TRUE): 'getMethods' is deprecated.</span></span>
<span><span class="co">#&gt; Use 'getMethodsForDispatch(f, TRUE)' instead.</span></span>
<span><span class="co">#&gt; See help("Deprecated")</span></span>
<span><span class="co">#&gt; Warning in getMethods(coerce, table = TRUE): 'getMethods' is deprecated.</span></span>
<span><span class="co">#&gt; Use 'getMethodsForDispatch(f, TRUE)' instead.</span></span>
<span><span class="co">#&gt; See help("Deprecated")</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>The wrapper function <code><a href="../reference/gghic.html">gghic()</a></code> can also add ideograms to
the plot. (exactly the same as above).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic_100</span><span class="op">[</span><span class="st">"chr11"</span><span class="op">]</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/gghic.html">gghic</a></span><span class="op">(</span></span>
<span>    ideogram <span class="op">=</span> <span class="cn">TRUE</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, highlight <span class="op">=</span> <span class="cn">FALSE</span>, length_ratio <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>    ideogram_fontsize <span class="op">=</span> <span class="fl">8</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>The plotted region can be highlighted on the ideogram by setting
<code>highlight = TRUE</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic_100</span><span class="op">[</span><span class="st">"chr4:10000000-15000000"</span><span class="op">]</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/gghic.html">gghic</a></span><span class="op">(</span></span>
<span>    ideogram <span class="op">=</span> <span class="cn">TRUE</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, highlight <span class="op">=</span> <span class="cn">TRUE</span>, length_ratio <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>    ideogram_fontsize <span class="op">=</span> <span class="fl">8</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="gene-annotation">Gene Annotation<a class="anchor" aria-label="anchor" href="#gene-annotation"></a>
</h3>
<p>Track-based gene annotation can be added to the plot by using
<code><a href="../reference/geom_annotation.html">geom_annotation()</a></code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">x_5</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">seqnames1</span> <span class="op">==</span> <span class="st">"chr11"</span>, <span class="va">seqnames2</span> <span class="op">==</span> <span class="st">"chr11"</span>,</span>
<span>    <span class="va">center1</span> <span class="op">&gt;</span> <span class="fl">67000000</span>, <span class="va">center1</span> <span class="op">&lt;</span> <span class="fl">67100000</span>,</span>
<span>    <span class="va">center2</span> <span class="op">&gt;</span> <span class="fl">67000000</span>, <span class="va">center2</span> <span class="op">&lt;</span> <span class="fl">67100000</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>      seqnames1 <span class="op">=</span> <span class="va">seqnames1</span>, start1 <span class="op">=</span> <span class="va">start1</span>, end1 <span class="op">=</span> <span class="va">end1</span>,</span>
<span>      seqnames2 <span class="op">=</span> <span class="va">seqnames2</span>, start2 <span class="op">=</span> <span class="va">start2</span>, end2 <span class="op">=</span> <span class="va">end2</span>,</span>
<span>      fill <span class="op">=</span> <span class="va">score</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_hic.html">geom_hic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/theme_hic.html">theme_hic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_ideogram.html">geom_ideogram</a></span><span class="op">(</span></span>
<span>    genome <span class="op">=</span> <span class="st">"hg19"</span>, highlight <span class="op">=</span> <span class="cn">FALSE</span>, length_ratio <span class="op">=</span> <span class="fl">0.7</span>, fontsize <span class="op">=</span> <span class="fl">8</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="../reference/geom_annotation.html">geom_annotation</a></span><span class="op">(</span>gtf_path <span class="op">=</span> <span class="va">path_gtf</span>, style <span class="op">=</span> <span class="st">"basic"</span>, maxgap <span class="op">=</span> <span class="fl">100000</span><span class="op">)</span></span>
<span><span class="co">#&gt; Import genomic features from the file as a GRanges object ... OK</span></span>
<span><span class="co">#&gt; Prepare the 'metadata' data frame ... OK</span></span>
<span><span class="co">#&gt; Make the TxDb object ...</span></span>
<span><span class="co">#&gt; Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The "phase" metadata column contains non-NA values for features of type</span></span>
<span><span class="co">#&gt;   stop_codon. This information was ignored.</span></span>
<span><span class="co">#&gt; Warning in .makeTxDb_normarg_chrominfo(chrominfo): genome version information</span></span>
<span><span class="co">#&gt; is not available for this TxDb object</span></span>
<span><span class="co">#&gt; OK</span></span>
<span><span class="co">#&gt; Warning in .makeTxDb_normarg_chrominfo(chrominfo): genome version information</span></span>
<span><span class="co">#&gt; is not available for this TxDb object</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>Some parameters can be set for <code><a href="../reference/geom_annotation.html">geom_annotation()</a></code>:</p>
<ul>
<li>
<code>style</code> can be set to <code>"arrow"</code> to change the
annotation style.</li>
<li>
<code>maxgap</code> is used to calculate how many non-overlapping
tracks are needed for the annotation.</li>
<li>
<code>gene_symbols</code> can be used to specify genes to plot.</li>
</ul>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="../reference/geom_annotation.html">geom_annotation</a></span><span class="op">(</span></span>
<span>  gtf_path <span class="op">=</span> <span class="va">path_gtf</span>, style <span class="op">=</span> <span class="st">"arrow"</span>, maxgap <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>,</span>
<span>  gene_symbols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GRK2"</span>, <span class="st">"SSH3"</span>, <span class="st">"KDM2A"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in .makeTxDb_normarg_chrominfo(chrominfo): genome version information</span></span>
<span><span class="co">#&gt; is not available for this TxDb object</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p><code><a href="../reference/gghic.html">gghic()</a></code> can also add gene annotation to the plot
(exactly the same as above).</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic_5</span><span class="op">[</span><span class="st">"chr11:67000000-67100000"</span><span class="op">]</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/gghic.html">gghic</a></span><span class="op">(</span></span>
<span>    ideogram <span class="op">=</span> <span class="cn">TRUE</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, highlight <span class="op">=</span> <span class="cn">FALSE</span>, length_ratio <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>    ideogram_fontsize <span class="op">=</span> <span class="fl">8</span>,</span>
<span></span>
<span>    annotation <span class="op">=</span> <span class="cn">TRUE</span>, gtf_path <span class="op">=</span> <span class="va">path_gtf</span>, style <span class="op">=</span> <span class="st">"arrow"</span>, maxgap <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>,</span>
<span>    gene_symbols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GRK2"</span>, <span class="st">"SSH3"</span>, <span class="st">"KDM2A"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="chromatin-loops">Chromatin Loops<a class="anchor" aria-label="anchor" href="#chromatin-loops"></a>
</h3>
<p>Chromatin loops can be visualised using <code><a href="../reference/geom_loop.html">geom_loop()</a></code> or
<code><a href="../reference/geom_loop2.html">geom_loop2()</a></code>. The former requires a path to a loop file,
while the latter requires a <code>tibble</code>/<code>data.frame</code>
containing the loop data which enables easy customization by adding
additional aesthetics.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic_5</span><span class="op">[</span><span class="st">"chr11:61915000-65000000"</span><span class="op">]</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/gghic.html">gghic</a></span><span class="op">(</span>loop <span class="op">=</span> <span class="cn">TRUE</span>, loop_path <span class="op">=</span> <span class="va">path_loop</span>, loop_is_0based <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df_loop</span> <span class="op">&lt;-</span> <span class="va">path_loop</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span></span>
<span>    sep <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>    col.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"seqnames1"</span>, <span class="st">"start1"</span>, <span class="st">"end1"</span>, <span class="st">"seqnames2"</span>, <span class="st">"start2"</span>, <span class="st">"end2"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">seqnames1</span> <span class="op">==</span> <span class="st">"chr11"</span>, <span class="va">seqnames2</span> <span class="op">==</span> <span class="st">"chr11"</span>,</span>
<span>    <span class="va">start1</span> <span class="op">&gt;</span> <span class="fl">61925000</span>, <span class="va">end1</span> <span class="op">&lt;</span> <span class="fl">67480000</span>,</span>
<span>    <span class="va">start2</span> <span class="op">&gt;</span> <span class="fl">61925000</span>, <span class="va">end2</span> <span class="op">&lt;</span> <span class="fl">67480000</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">df_loop</span><span class="op">)</span>, <span class="fl">6</span><span class="op">)</span></span>
<span><span class="va">df_loop</span> <span class="op">&lt;-</span> <span class="va">df_loop</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="va">keep</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="fl">3</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"B"</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">hic_5</span><span class="op">[</span><span class="st">"chr11:61915000-65000000"</span><span class="op">]</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/gghic.html">gghic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_loop2.html">geom_loop2</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">df_loop</span>,</span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>      seqnames1 <span class="op">=</span> <span class="va">seqnames1</span>, start1 <span class="op">=</span> <span class="va">start1</span>, end1 <span class="op">=</span> <span class="va">end1</span>,</span>
<span>      seqnames2 <span class="op">=</span> <span class="va">seqnames2</span>, start2 <span class="op">=</span> <span class="va">start2</span>, end2 <span class="op">=</span> <span class="va">end2</span>,</span>
<span>      colour <span class="op">=</span> <span class="va">sample</span></span>
<span>    <span class="op">)</span>,</span>
<span>    stroke <span class="op">=</span> <span class="fl">1</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="tads">TADs<a class="anchor" aria-label="anchor" href="#tads"></a>
</h3>
<p>Triangulers indicating TADs can be added using
<code><a href="../reference/geom_tad.html">geom_tad()</a></code> or <code><a href="../reference/geom_tad2.html">geom_tad2()</a></code>. The former requires
a path to a TAD file, while the latter requires a
<code>tibble</code>/<code>data.frame</code> containing the TAD data.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x_100</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">seqnames1</span> <span class="op">==</span> <span class="st">"chr4"</span>, <span class="va">seqnames2</span> <span class="op">==</span> <span class="st">"chr4"</span>,</span>
<span>    <span class="va">start1</span> <span class="op">&gt;</span> <span class="fl">50000000</span>, <span class="va">end1</span> <span class="op">&lt;</span> <span class="fl">80000000</span>,</span>
<span>    <span class="va">start2</span> <span class="op">&gt;</span> <span class="fl">50000000</span>, <span class="va">end2</span> <span class="op">&lt;</span> <span class="fl">80000000</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/gghic.html">gghic</a></span><span class="op">(</span></span>
<span>    tad <span class="op">=</span> <span class="cn">TRUE</span>, tad_is_0based <span class="op">=</span> <span class="cn">TRUE</span>, tad_path <span class="op">=</span> <span class="va">path_tad</span>, tad_colour <span class="op">=</span> <span class="st">"#00ff83"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df_tad</span> <span class="op">&lt;-</span> <span class="va">path_tad</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span></span>
<span>    sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">FALSE</span>, col.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"seqnames"</span>, <span class="st">"start"</span>, <span class="st">"end"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>start <span class="op">=</span> <span class="va">start</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">seqnames</span> <span class="op">==</span> <span class="st">"chr4"</span>, <span class="va">start</span> <span class="op">&gt;</span> <span class="fl">60000000</span>, <span class="va">end</span> <span class="op">&lt;</span> <span class="fl">70000000</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">x_100</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">seqnames1</span> <span class="op">==</span> <span class="st">"chr4"</span>, <span class="va">seqnames2</span> <span class="op">==</span> <span class="st">"chr4"</span>,</span>
<span>    <span class="va">start1</span> <span class="op">&gt;</span> <span class="fl">60000000</span>, <span class="va">end1</span> <span class="op">&lt;</span> <span class="fl">70000000</span>,</span>
<span>    <span class="va">start2</span> <span class="op">&gt;</span> <span class="fl">60000000</span>, <span class="va">end2</span> <span class="op">&lt;</span> <span class="fl">70000000</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/gghic.html">gghic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_tad2.html">geom_tad2</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">df_tad</span>,</span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>      seqnames <span class="op">=</span> <span class="va">seqnames</span>, start <span class="op">=</span> <span class="va">start</span>, end <span class="op">=</span> <span class="va">end</span>,</span>
<span>      colour <span class="op">=</span> <span class="va">sample</span></span>
<span>    <span class="op">)</span>,</span>
<span>    stroke <span class="op">=</span> <span class="fl">2</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_grey.html" class="external-link">scale_color_grey</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="multi-way-contacts">Multi-way contacts<a class="anchor" aria-label="anchor" href="#multi-way-contacts"></a>
</h3>
<p>Multi-way contacts from concatemers obtained from Pore-C data can be
visualised using <code><a href="../reference/geom_concatemer.html">geom_concatemer()</a></code> or
<code><a href="../reference/geom_concatemer2.html">geom_concatemer2()</a></code>. The former requires a path to a
concatemer file, while the latter requires a
<code>tibble</code>/<code>data.frame</code> containing the concatemer
data.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gis</span> <span class="op">&lt;-</span> <span class="va">path_gis_hic</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">scale_data</span><span class="op">(</span><span class="st">"balanced"</span>, <span class="va">log10</span><span class="op">)</span></span>
<span></span>
<span><span class="va">names_read</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"0c32bbce-24f8-4006-a75c-0e3504648971"</span>,</span>
<span>  <span class="st">"10ae8263-cebc-4cbc-94c1-2c1dc5f22101"</span>,</span>
<span>  <span class="st">"ced078ca-2d68-5217-a20e-7dc37dd2d786"</span>,</span>
<span>  <span class="st">"06063b40-0271-4f4e-993d-2f202028b697"</span>,</span>
<span>  <span class="st">"08bb8fe4-c54b-4a11-b9a0-c9eed2d18933"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">df_concatemers</span> <span class="op">&lt;-</span> <span class="va">path_concatemers</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu">sort</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">read_name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">names_read</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/gghic.html">gghic</a></span><span class="op">(</span><span class="va">gis</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="../reference/geom_concatemer2.html">geom_concatemer2</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">df_concatemers</span>,</span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>      seqnames <span class="op">=</span> <span class="va">seqnames</span>, start <span class="op">=</span> <span class="va">start</span>, end <span class="op">=</span> <span class="va">end</span>, read_group <span class="op">=</span> <span class="va">read_name</span>,</span>
<span>      colour <span class="op">=</span> <span class="va">read_name</span></span>
<span>    <span class="op">)</span>,</span>
<span>    check_concatemers <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>    legend.key.size <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html" class="external-link">unit</a></span><span class="op">(</span><span class="fl">2</span>, <span class="st">"mm"</span><span class="op">)</span>,</span>
<span>    legend.text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="visualisation-of-interaction-data-of-multiple-chromosomes">Visualisation of Interaction Data of Multiple Chromosomes<a class="anchor" aria-label="anchor" href="#visualisation-of-interaction-data-of-multiple-chromosomes"></a>
</h3>
<p>Set <code>draw_boundary</code> to <code>TRUE</code> to draw the
boundary of the chromosomes. Set <code>expand_xaxis</code> to
<code>TRUE</code> to expand the x-axis if the plot is crowded.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x_100</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/gghic.html">gghic</a></span><span class="op">(</span></span>
<span>    draw_boundary <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    expand_xaxis <span class="op">=</span> <span class="cn">TRUE</span>, expand_left <span class="op">=</span> <span class="fl">2000000</span>, expand_right <span class="op">=</span> <span class="fl">2000000</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-21-1.png" width="700"></p>
<p>Chromosome ideograms and gene tracks can also be added to the plot
when visualising multiple chromosomes.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">x_5</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">center1</span> <span class="op">&gt;</span> <span class="fl">10000000</span> <span class="op">&amp;</span> <span class="va">center1</span> <span class="op">&lt;</span> <span class="fl">11000000</span> <span class="op">&amp;</span></span>
<span>      <span class="va">center2</span> <span class="op">&gt;</span> <span class="fl">10000000</span> <span class="op">&amp;</span> <span class="va">center2</span> <span class="op">&lt;</span> <span class="fl">11000000</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/gghic.html">gghic</a></span><span class="op">(</span></span>
<span>    draw_boundary <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span></span>
<span>    ideogram <span class="op">=</span> <span class="cn">TRUE</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, highlight <span class="op">=</span> <span class="cn">TRUE</span>, ideogram_fontsize <span class="op">=</span> <span class="fl">7</span>,</span>
<span>    ideogram_width_ratio <span class="op">=</span> <span class="fl">0.08</span>,</span>
<span></span>
<span>    annotation <span class="op">=</span> <span class="cn">TRUE</span>, include_ncrna <span class="op">=</span> <span class="cn">FALSE</span>, gtf_path <span class="op">=</span> <span class="va">path_gtf</span>,</span>
<span>    style <span class="op">=</span> <span class="st">"basic"</span>, maxgap <span class="op">=</span> <span class="fl">100000</span>, annotation_fontsize <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    annotation_width_ratio <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span></span>
<span>    expand_xaxis <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span>
<span><span class="co">#&gt; Warning in .makeTxDb_normarg_chrominfo(chrominfo): genome version information</span></span>
<span><span class="co">#&gt; is not available for this TxDb object</span></span>
<span><span class="co">#&gt; Warning in .makeTxDb_normarg_chrominfo(chrominfo): genome version information</span></span>
<span><span class="co">#&gt; is not available for this TxDb object</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-22-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="other-data-tracks---geom_track">Other Data Tracks - <code>geom_track</code><a class="anchor" aria-label="anchor" href="#other-data-tracks---geom_track"></a>
</h3>
<p><code><a href="../reference/geom_track.html">geom_track()</a></code> can be used to add other data tracks to the
plot.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="../reference/geom_track.html">geom_track</a></span><span class="op">(</span></span>
<span>  data_paths <span class="op">=</span> <span class="va">paths_track</span>, width_ratio <span class="op">=</span> <span class="fl">0.3</span>, data_range <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#DC0000B2"</span>, <span class="st">"#00A087B2"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in .makeTxDb_normarg_chrominfo(chrominfo): genome version information</span></span>
<span><span class="co">#&gt; is not available for this TxDb object</span></span>
<span><span class="co">#&gt; Warning in .makeTxDb_normarg_chrominfo(chrominfo): genome version information</span></span>
<span><span class="co">#&gt; is not available for this TxDb object</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-23-1.png" width="700"></p>
<p><code><a href="../reference/gghic.html">gghic()</a></code> is handy for adding all these features to the
plot.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x_5</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>    <span class="va">center1</span> <span class="op">&gt;</span> <span class="fl">10000000</span> <span class="op">&amp;</span> <span class="va">center1</span> <span class="op">&lt;</span> <span class="fl">11000000</span> <span class="op">&amp;</span></span>
<span>      <span class="va">center2</span> <span class="op">&gt;</span> <span class="fl">10000000</span> <span class="op">&amp;</span> <span class="va">center2</span> <span class="op">&lt;</span> <span class="fl">11000000</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/gghic.html">gghic</a></span><span class="op">(</span></span>
<span>    draw_boundary <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span></span>
<span>    ideogram <span class="op">=</span> <span class="cn">TRUE</span>, genome <span class="op">=</span> <span class="st">"hg19"</span>, highlight <span class="op">=</span> <span class="cn">TRUE</span>, ideogram_fontsize <span class="op">=</span> <span class="fl">7</span>,</span>
<span>    ideogram_width_ratio <span class="op">=</span> <span class="fl">0.08</span>,</span>
<span>    annotation <span class="op">=</span> <span class="cn">TRUE</span>, include_ncrna <span class="op">=</span> <span class="cn">FALSE</span>, gtf_path <span class="op">=</span> <span class="va">path_gtf</span>,</span>
<span>    style <span class="op">=</span> <span class="st">"arrow"</span>, maxgap <span class="op">=</span> <span class="fl">100000</span>, annotation_fontsize <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    annotation_width_ratio <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span></span>
<span>    track <span class="op">=</span> <span class="cn">TRUE</span>, data_paths <span class="op">=</span> <span class="va">paths_track</span>, track_width_ratio <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>    track_fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#DC0000B2"</span>, <span class="st">"#00A087B2"</span><span class="op">)</span>, data_range <span class="op">=</span> <span class="st">"maximum"</span>,</span>
<span></span>
<span>    expand_xaxis <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; Warning in .makeTxDb_normarg_chrominfo(chrominfo): genome version information</span></span>
<span><span class="co">#&gt; is not available for this TxDb object</span></span>
<span><span class="co">#&gt; Warning in .makeTxDb_normarg_chrominfo(chrominfo): genome version information</span></span>
<span><span class="co">#&gt; is not available for this TxDb object</span></span></code></pre></div>
<p><img src="gghic_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.0 (2025-04-11)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] GenomicFeatures_1.60.0      AnnotationDbi_1.70.0       </span></span>
<span><span class="co">#&gt;  [3] InteractionSet_1.36.1       SummarizedExperiment_1.38.1</span></span>
<span><span class="co">#&gt;  [5] Biobase_2.68.0              MatrixGenerics_1.20.0      </span></span>
<span><span class="co">#&gt;  [7] matrixStats_1.5.0           GenomicRanges_1.60.0       </span></span>
<span><span class="co">#&gt;  [9] GenomeInfoDb_1.44.0         IRanges_2.42.0             </span></span>
<span><span class="co">#&gt; [11] S4Vectors_0.46.0            BiocGenerics_0.54.0        </span></span>
<span><span class="co">#&gt; [13] generics_0.1.4              HiCExperiment_1.8.0        </span></span>
<span><span class="co">#&gt; [15] glue_1.8.0                  tidyr_1.3.1                </span></span>
<span><span class="co">#&gt; [17] dplyr_1.1.4                 scales_1.4.0               </span></span>
<span><span class="co">#&gt; [19] tibble_3.2.1                ggplot2_3.5.2              </span></span>
<span><span class="co">#&gt; [21] gghic_0.1.0                </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] strawr_0.0.92            RColorBrewer_1.1-3       rstudioapi_0.17.1       </span></span>
<span><span class="co">#&gt;   [4] jsonlite_2.0.0           magrittr_2.0.3           farver_2.1.2            </span></span>
<span><span class="co">#&gt;   [7] rmarkdown_2.29           fs_1.6.6                 BiocIO_1.18.0           </span></span>
<span><span class="co">#&gt;  [10] ragg_1.4.0               vctrs_0.6.5              memoise_2.0.1           </span></span>
<span><span class="co">#&gt;  [13] Rsamtools_2.24.0         RCurl_1.98-1.17          base64enc_0.1-3         </span></span>
<span><span class="co">#&gt;  [16] htmltools_0.5.8.1        S4Arrays_1.8.0           progress_1.2.3          </span></span>
<span><span class="co">#&gt;  [19] curl_6.2.2               Rhdf5lib_1.30.0          rhdf5_2.52.0            </span></span>
<span><span class="co">#&gt;  [22] SparseArray_1.8.0        Formula_1.2-5            sass_0.4.10             </span></span>
<span><span class="co">#&gt;  [25] bslib_0.9.0              htmlwidgets_1.6.4        desc_1.4.3              </span></span>
<span><span class="co">#&gt;  [28] Gviz_1.52.0              httr2_1.1.2              cachem_1.1.0            </span></span>
<span><span class="co">#&gt;  [31] GenomicAlignments_1.44.0 lifecycle_1.0.4          pkgconfig_2.0.3         </span></span>
<span><span class="co">#&gt;  [34] Matrix_1.7-3             R6_2.6.1                 fastmap_1.2.0           </span></span>
<span><span class="co">#&gt;  [37] GenomeInfoDbData_1.2.14  digest_0.6.37            colorspace_2.1-1        </span></span>
<span><span class="co">#&gt;  [40] rprojroot_2.0.4          textshaping_1.0.1        Hmisc_5.2-3             </span></span>
<span><span class="co">#&gt;  [43] RSQLite_2.3.11           labeling_0.4.3           filelock_1.0.3          </span></span>
<span><span class="co">#&gt;  [46] httr_1.4.7               abind_1.4-8              compiler_4.5.0          </span></span>
<span><span class="co">#&gt;  [49] here_1.0.1               withr_3.0.2              bit64_4.6.0-1           </span></span>
<span><span class="co">#&gt;  [52] htmlTable_2.4.3          backports_1.5.0          BiocParallel_1.42.0     </span></span>
<span><span class="co">#&gt;  [55] DBI_1.2.3                biomaRt_2.64.0           rappdirs_0.3.3          </span></span>
<span><span class="co">#&gt;  [58] DelayedArray_0.34.1      rjson_0.2.23             tools_4.5.0             </span></span>
<span><span class="co">#&gt;  [61] foreign_0.8-90           nnet_7.3-20              restfulr_0.0.15         </span></span>
<span><span class="co">#&gt;  [64] rhdf5filters_1.20.0      grid_4.5.0               checkmate_2.3.2         </span></span>
<span><span class="co">#&gt;  [67] cluster_2.1.8.1          gtable_0.3.6             BSgenome_1.76.0         </span></span>
<span><span class="co">#&gt;  [70] tzdb_0.5.0               ensembldb_2.32.0         data.table_1.17.2       </span></span>
<span><span class="co">#&gt;  [73] hms_1.1.3                xml2_1.3.8               XVector_0.48.0          </span></span>
<span><span class="co">#&gt;  [76] pillar_1.10.2            stringr_1.5.1            vroom_1.6.5             </span></span>
<span><span class="co">#&gt;  [79] BiocFileCache_2.16.0     lattice_0.22-6           rtracklayer_1.68.0      </span></span>
<span><span class="co">#&gt;  [82] bit_4.6.0                deldir_2.0-4             biovizBase_1.56.0       </span></span>
<span><span class="co">#&gt;  [85] tidyselect_1.2.1         Biostrings_2.76.0        knitr_1.50              </span></span>
<span><span class="co">#&gt;  [88] gridExtra_2.3            ProtGenerics_1.40.0      xfun_0.52               </span></span>
<span><span class="co">#&gt;  [91] stringi_1.8.7            UCSC.utils_1.4.0         lazyeval_0.2.2          </span></span>
<span><span class="co">#&gt;  [94] yaml_2.3.10              evaluate_1.0.3           codetools_0.2-20        </span></span>
<span><span class="co">#&gt;  [97] interp_1.1-6             cli_3.6.5                rpart_4.1.24            </span></span>
<span><span class="co">#&gt; [100] reticulate_1.42.0        systemfonts_1.2.3        jquerylib_0.1.4         </span></span>
<span><span class="co">#&gt; [103] dichromat_2.0-0.1        Rcpp_1.0.14              dbplyr_2.5.0            </span></span>
<span><span class="co">#&gt; [106] png_0.1-8                XML_3.99-0.18            parallel_4.5.0          </span></span>
<span><span class="co">#&gt; [109] pkgdown_2.1.2            blob_1.2.4               prettyunits_1.2.0       </span></span>
<span><span class="co">#&gt; [112] latticeExtra_0.6-30      jpeg_0.1-11              AnnotationFilter_1.32.0 </span></span>
<span><span class="co">#&gt; [115] bitops_1.0-9             txdbmaker_1.4.1          VariantAnnotation_1.54.1</span></span>
<span><span class="co">#&gt; [118] purrr_1.0.4              crayon_1.5.3             rlang_1.1.6             </span></span>
<span><span class="co">#&gt; [121] KEGGREST_1.48.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Minghao Jiang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
